--- a/util.c
+++ b/util.c
@@ -8,7 +8,7 @@
 #include "util.h"
 
 void
-die(const char* fmt, ...)
+die(const char* const fmt, ...)
 {
 	va_list ap;
 	int saved_errno;
--- a/drw.c
+++ b/drw.c
@@ -13,6 +13,9 @@
 static int
 utf8decode(const char* s_in, long* u, int* err)
 {
+	if (!s_in || !u || !err)
+		return 0;
+
 	static const unsigned char lens[] = {
 		// clang-format off
 		/* 0XXXX */ 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
@@ -373,6 +376,7 @@ drw_text(Drw* drw, int x, int y, unsigne
 
 			if (!drw->fonts || !drw->fonts->pattern) {
 				/* Refer to the comment in xfont_create for more information. */
+				FcCharSetDestroy(fccharset);
 				die("the first font in the cache must be loaded from a font string.");
 				return 0;
 			}
