--- a/config.def.h
+++ b/config.def.h
@@ -1,26 +1,27 @@
 /* See LICENSE file for copyright and license details. */
 
 /* appearance */
-static const unsigned int borderpx  = 1;        /* border pixel of windows */
-static const unsigned int snap      = 32;       /* snap pixel */
-static const int showbar            = 1;        /* 0 means no bar */
-static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
-static const char col_gray1[]       = "#222222";
-static const char col_gray2[]       = "#444444";
-static const char col_gray3[]       = "#bbbbbb";
-static const char col_gray4[]       = "#eeeeee";
-static const char col_cyan[]        = "#005577";
-static const char *colors[][3]      = {
-	/*               fg         bg         border   */
-	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
-	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
+static const unsigned int borderpx = 1;  /* border pixel of windows */
+static const unsigned int snap     = 32; /* snap pixel */
+static const int showbar           = 1;  /* 0 means no bar */
+static const int topbar            = 1;  /* 0 means bottom bar */
+static const char* fonts[]         = { "monospace:size=10" };
+static const char dmenufont[]      = "monospace:size=10";
+static const char col_gray1[]      = "#222222";
+static const char col_gray2[]      = "#444444";
+static const char col_gray3[]      = "#bbbbbb";
+static const char col_gray4[]      = "#eeeeee";
+static const char col_cyan[]       = "#005577";
+static const char* colors[][3]     = {
+    /*               fg         bg         border   */
+    [SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
+    [SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char* tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
 
+// clang-format off
 static const Rule rules[] = {
 	/* xprop(1):
 	 *	WM_CLASS(STRING) = instance, class
@@ -30,24 +31,25 @@ static const Rule rules[] = {
 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
 	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
 };
+// clang-format on
 
 /* layout(s) */
-static const float mfact     = 0.55; /* factor of master area size [0.05..0.95] */
-static const int nmaster     = 1;    /* number of clients in master area */
-static const int resizehints = 1;    /* 1 means respect size hints in tiled resizals */
-static const int lockfullscreen = 1; /* 1 will force focus on the fullscreen window */
-static const int refreshrate = 120;  /* refresh rate (per second) for client move/resize */
+static const float mfact        = 0.55; /* factor of master area size [0.05..0.95] */
+static const int nmaster        = 1;    /* number of clients in master area */
+static const int resizehints    = 1;    /* 1 means respect size hints in tiled resizals */
+static const int lockfullscreen = 1;    /* 1 will force focus on the fullscreen window */
+static const int refreshrate    = 120;  /* refresh rate (per second) for client move/resize */
 
 static const Layout layouts[] = {
-	/* symbol     arrange function */
-	{ "[]=",      tile },    /* first entry is default */
-	{ "><>",      NULL },    /* no layout function means floating behavior */
-	{ "[M]",      monocle },
+	/* symbol  arrange function */
+	{ "[]=", tile    }, /* first entry is default */
+	{ "><>", NULL    }, /* no layout function means floating behavior */
+	{ "[M]", monocle },
 };
 
 /* key definitions */
 #define MODKEY Mod1Mask
-#define TAGKEYS(KEY,TAG) \
+#define TAGKEYS(KEY, TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
 	{ MODKEY|ShiftMask,             KEY,      tag,            {.ui = 1 << TAG} }, \
@@ -57,61 +59,62 @@ static const Layout layouts[] = {
 #define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }
 
 /* commands */
-static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
-static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static char dmenumon[2]       = "0"; /* component of dmenucmd, manipulated in spawn() */
+static const char* dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
+static const char* termcmd[]  = { "st", NULL };
 
 static const Key keys[] = {
-	/* modifier                     key        function        argument */
-	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
-	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
-	{ MODKEY,                       XK_b,      togglebar,      {0} },
-	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
-	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
-	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
-	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
-	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
-	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
-	{ MODKEY,                       XK_Return, zoom,           {0} },
-	{ MODKEY,                       XK_Tab,    view,           {0} },
-	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
-	{ MODKEY,                       XK_t,      setlayout,      {.v = &layouts[0]} },
-	{ MODKEY,                       XK_f,      setlayout,      {.v = &layouts[1]} },
-	{ MODKEY,                       XK_m,      setlayout,      {.v = &layouts[2]} },
-	{ MODKEY,                       XK_space,  setlayout,      {0} },
-	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
-	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
-	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
-	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
-	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
-	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
-	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
-	TAGKEYS(                        XK_1,                      0)
-	TAGKEYS(                        XK_2,                      1)
-	TAGKEYS(                        XK_3,                      2)
-	TAGKEYS(                        XK_4,                      3)
-	TAGKEYS(                        XK_5,                      4)
-	TAGKEYS(                        XK_6,                      5)
-	TAGKEYS(                        XK_7,                      6)
-	TAGKEYS(                        XK_8,                      7)
-	TAGKEYS(                        XK_9,                      8)
-	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	/* modifier           key        function        argument */
+	{ MODKEY,             XK_p,      spawn,          { .v = dmenucmd }    },
+	{ MODKEY | ShiftMask, XK_Return, spawn,          { .v = termcmd }     },
+	{ MODKEY,             XK_b,      togglebar,      { 0 }                },
+	{ MODKEY,             XK_j,      focusstack,     { .i = +1 }          },
+	{ MODKEY,             XK_k,      focusstack,     { .i = -1 }          },
+	{ MODKEY,             XK_i,      incnmaster,     { .i = +1 }          },
+	{ MODKEY,             XK_d,      incnmaster,     { .i = -1 }          },
+	{ MODKEY,             XK_h,      setmfact,       { .f = -0.05 }       },
+	{ MODKEY,             XK_l,      setmfact,       { .f = +0.05 }       },
+	{ MODKEY,             XK_Return, zoom,           { 0 }                },
+	{ MODKEY,             XK_Tab,    view,           { 0 }                },
+	{ MODKEY | ShiftMask, XK_c,      killclient,     { 0 }                },
+	{ MODKEY,             XK_t,      setlayout,      { .v = &layouts[0] } },
+	{ MODKEY,             XK_f,      setlayout,      { .v = &layouts[1] } },
+	{ MODKEY,             XK_m,      setlayout,      { .v = &layouts[2] } },
+	{ MODKEY,             XK_space,  setlayout,      { 0 }                },
+	{ MODKEY | ShiftMask, XK_space,  togglefloating, { 0 }                },
+	{ MODKEY,             XK_0,      view,           { .ui = ~0 }         },
+	{ MODKEY | ShiftMask, XK_0,      tag,            { .ui = ~0 }         },
+	{ MODKEY,             XK_comma,  focusmon,       { .i = -1 }          },
+	{ MODKEY,             XK_period, focusmon,       { .i = +1 }          },
+	{ MODKEY | ShiftMask, XK_comma,  tagmon,         { .i = -1 }          },
+	{ MODKEY | ShiftMask, XK_period, tagmon,         { .i = +1 }          },
+	// clang-format off
+	TAGKEYS(              XK_1,                      0)
+	TAGKEYS(              XK_2,                      1)
+	TAGKEYS(              XK_3,                      2)
+	TAGKEYS(              XK_4,                      3)
+	TAGKEYS(              XK_5,                      4)
+	TAGKEYS(              XK_6,                      5)
+	TAGKEYS(              XK_7,                      6)
+	TAGKEYS(              XK_8,                      7)
+	TAGKEYS(              XK_9,                      8)
+	// clang-format on
+	{ MODKEY | ShiftMask, XK_q,      quit,           { 0 }                },
 };
 
 /* button definitions */
 /* click can be ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle, ClkClientWin, or ClkRootWin */
 static const Button buttons[] = {
-	/* click                event mask      button          function        argument */
-	{ ClkLtSymbol,          0,              Button1,        setlayout,      {0} },
-	{ ClkLtSymbol,          0,              Button3,        setlayout,      {.v = &layouts[2]} },
-	{ ClkWinTitle,          0,              Button2,        zoom,           {0} },
-	{ ClkStatusText,        0,              Button2,        spawn,          {.v = termcmd } },
-	{ ClkClientWin,         MODKEY,         Button1,        movemouse,      {0} },
-	{ ClkClientWin,         MODKEY,         Button2,        togglefloating, {0} },
-	{ ClkClientWin,         MODKEY,         Button3,        resizemouse,    {0} },
-	{ ClkTagBar,            0,              Button1,        view,           {0} },
-	{ ClkTagBar,            0,              Button3,        toggleview,     {0} },
-	{ ClkTagBar,            MODKEY,         Button1,        tag,            {0} },
-	{ ClkTagBar,            MODKEY,         Button3,        toggletag,      {0} },
+	/* click         mask    button   function        argument */
+	{ ClkLtSymbol,   0,      Button1, setlayout,      { 0 }                },
+	{ ClkLtSymbol,   0,      Button3, setlayout,      { .v = &layouts[2] } },
+	{ ClkWinTitle,   0,      Button2, zoom,           { 0 }                },
+	{ ClkStatusText, 0,      Button2, spawn,          { .v = termcmd }     },
+	{ ClkClientWin,  MODKEY, Button1, movemouse,      { 0 }                },
+	{ ClkClientWin,  MODKEY, Button2, togglefloating, { 0 }                },
+	{ ClkClientWin,  MODKEY, Button3, resizemouse,    { 0 }                },
+	{ ClkTagBar,     0,      Button1, view,           { 0 }                },
+	{ ClkTagBar,     0,      Button3, toggleview,     { 0 }                },
+	{ ClkTagBar,     MODKEY, Button1, tag,            { 0 }                },
+	{ ClkTagBar,     MODKEY, Button3, toggletag,      { 0 }                },
 };
-
--- a/drw.c
+++ b/drw.c
@@ -2,8 +2,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <X11/Xlib.h>
 #include <X11/Xft/Xft.h>
+#include <X11/Xlib.h>
 
 #include "drw.h"
 #include "util.h"
@@ -11,23 +11,25 @@
 #define UTF_INVALID 0xFFFD
 
 static int
-utf8decode(const char *s_in, long *u, int *err)
+utf8decode(const char* s_in, long* u, int* err)
 {
 	static const unsigned char lens[] = {
+		// clang-format off
 		/* 0XXXX */ 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-		/* 10XXX */ 0, 0, 0, 0, 0, 0, 0, 0,  /* invalid */
+		/* 10XXX */ 0, 0, 0, 0, 0, 0, 0, 0, /* invalid */
 		/* 110XX */ 2, 2, 2, 2,
 		/* 1110X */ 3, 3,
 		/* 11110 */ 4,
-		/* 11111 */ 0,  /* invalid */
+		/* 11111 */ 0, /* invalid */
+		// clang-format on
 	};
 	static const unsigned char leading_mask[] = { 0x7F, 0x1F, 0x0F, 0x07 };
-	static const unsigned int overlong[] = { 0x0, 0x80, 0x0800, 0x10000 };
+	static const unsigned int overlong[]      = { 0x0, 0x80, 0x0800, 0x10000 };
 
-	const unsigned char *s = (const unsigned char *)s_in;
-	int len = lens[*s >> 3];
-	*u = UTF_INVALID;
-	*err = 1;
+	const unsigned char* s = (const unsigned char*) s_in;
+	int len                = lens[*s >> 3];
+	*u                     = UTF_INVALID;
+	*err                   = 1;
 	if (len == 0)
 		return 1;
 
@@ -42,29 +44,29 @@ utf8decode(const char *s_in, long *u, in
 		return len;
 
 	*err = 0;
-	*u = cp;
+	*u   = cp;
 	return len;
 }
 
-Drw *
-drw_create(Display *dpy, int screen, Window root, unsigned int w, unsigned int h)
+Drw*
+drw_create(Display* dpy, int screen, Window root, unsigned int w, unsigned int h)
 {
-	Drw *drw = ecalloc(1, sizeof(Drw));
+	Drw* drw = ecalloc(1, sizeof(Drw));
 
-	drw->dpy = dpy;
-	drw->screen = screen;
-	drw->root = root;
-	drw->w = w;
-	drw->h = h;
+	drw->dpy      = dpy;
+	drw->screen   = screen;
+	drw->root     = root;
+	drw->w        = w;
+	drw->h        = h;
 	drw->drawable = XCreatePixmap(dpy, root, w, h, DefaultDepth(dpy, screen));
-	drw->gc = XCreateGC(dpy, root, 0, NULL);
+	drw->gc       = XCreateGC(dpy, root, 0, NULL);
 	XSetLineAttributes(dpy, drw->gc, 1, LineSolid, CapButt, JoinMiter);
 
 	return drw;
 }
 
 void
-drw_resize(Drw *drw, unsigned int w, unsigned int h)
+drw_resize(Drw* drw, unsigned int w, unsigned int h)
 {
 	if (!drw)
 		return;
@@ -77,7 +79,7 @@ drw_resize(Drw *drw, unsigned int w, uns
 }
 
 void
-drw_free(Drw *drw)
+drw_free(Drw* drw)
 {
 	XFreePixmap(drw->dpy, drw->drawable);
 	XFreeGC(drw->dpy, drw->gc);
@@ -88,12 +90,12 @@ drw_free(Drw *drw)
 /* This function is an implementation detail. Library users should use
  * drw_fontset_create instead.
  */
-static Fnt *
-xfont_create(Drw *drw, const char *fontname, FcPattern *fontpattern)
+static Fnt*
+xfont_create(Drw* drw, const char* fontname, FcPattern* fontpattern)
 {
-	Fnt *font;
-	XftFont *xfont = NULL;
-	FcPattern *pattern = NULL;
+	Fnt* font;
+	XftFont* xfont     = NULL;
+	FcPattern* pattern = NULL;
 
 	if (fontname) {
 		/* Using the pattern found at font->xfont->pattern does not yield the
@@ -105,7 +107,7 @@ xfont_create(Drw *drw, const char *fontn
 			fprintf(stderr, "error, cannot load font from name: '%s'\n", fontname);
 			return NULL;
 		}
-		if (!(pattern = FcNameParse((FcChar8 *) fontname))) {
+		if (!(pattern = FcNameParse((FcChar8*) fontname))) {
 			fprintf(stderr, "error, cannot parse font name to pattern: '%s'\n", fontname);
 			XftFontClose(drw->dpy, xfont);
 			return NULL;
@@ -119,17 +121,17 @@ xfont_create(Drw *drw, const char *fontn
 		die("no font specified.");
 	}
 
-	font = ecalloc(1, sizeof(Fnt));
-	font->xfont = xfont;
+	font          = ecalloc(1, sizeof(Fnt));
+	font->xfont   = xfont;
 	font->pattern = pattern;
-	font->h = xfont->ascent + xfont->descent;
-	font->dpy = drw->dpy;
+	font->h       = xfont->ascent + xfont->descent;
+	font->dpy     = drw->dpy;
 
 	return font;
 }
 
 static void
-xfont_free(Fnt *font)
+xfont_free(Fnt* font)
 {
 	if (!font)
 		return;
@@ -140,7 +142,7 @@ xfont_free(Fnt *font)
 }
 
 Fnt*
-drw_fontset_create(Drw* drw, const char *fonts[], size_t fontcount)
+drw_fontset_create(Drw* drw, const char* fonts[], size_t fontcount)
 {
 	Fnt *cur, *ret = NULL;
 	size_t i;
@@ -151,14 +153,14 @@ drw_fontset_create(Drw* drw, const char
 	for (i = 1; i <= fontcount; i++) {
 		if ((cur = xfont_create(drw, fonts[fontcount - i], NULL))) {
 			cur->next = ret;
-			ret = cur;
+			ret       = cur;
 		}
 	}
 	return (drw->fonts = ret);
 }
 
 void
-drw_fontset_free(Fnt *font)
+drw_fontset_free(Fnt* font)
 {
 	if (font) {
 		drw_fontset_free(font->next);
@@ -167,7 +169,7 @@ drw_fontset_free(Fnt *font)
 }
 
 void
-drw_clr_create(Drw *drw, Clr *dest, const char *clrname)
+drw_clr_create(Drw* drw, Clr* dest, const char* clrname)
 {
 	if (!drw || !dest || !clrname)
 		return;
@@ -179,11 +181,11 @@ drw_clr_create(Drw *drw, Clr *dest, cons
 }
 
 /* Create color schemes. */
-Clr *
-drw_scm_create(Drw *drw, const char *clrnames[], size_t clrcount)
+Clr*
+drw_scm_create(Drw* drw, const char* clrnames[], size_t clrcount)
 {
 	size_t i;
-	Clr *ret;
+	Clr* ret;
 
 	/* need at least two colors for a scheme */
 	if (!drw || !clrnames || clrcount < 2 || !(ret = ecalloc(clrcount, sizeof(Clr))))
@@ -195,7 +197,7 @@ drw_scm_create(Drw *drw, const char *clr
 }
 
 void
-drw_clr_free(Drw *drw, Clr *c)
+drw_clr_free(Drw* drw, Clr* c)
 {
 	if (!drw || !c)
 		return;
@@ -206,7 +208,7 @@ drw_clr_free(Drw *drw, Clr *c)
 }
 
 void
-drw_scm_free(Drw *drw, Clr *scm, size_t clrcount)
+drw_scm_free(Drw* drw, Clr* scm, size_t clrcount)
 {
 	size_t i;
 
@@ -219,21 +221,21 @@ drw_scm_free(Drw *drw, Clr *scm, size_t
 }
 
 void
-drw_setfontset(Drw *drw, Fnt *set)
+drw_setfontset(Drw* drw, Fnt* set)
 {
 	if (drw)
 		drw->fonts = set;
 }
 
 void
-drw_setscheme(Drw *drw, Clr *scm)
+drw_setscheme(Drw* drw, Clr* scm)
 {
 	if (drw)
 		drw->scheme = scm;
 }
 
 void
-drw_rect(Drw *drw, int x, int y, unsigned int w, unsigned int h, int filled, int invert)
+drw_rect(Drw* drw, int x, int y, unsigned int w, unsigned int h, int filled, int invert)
 {
 	if (!drw || !drw->scheme)
 		return;
@@ -245,18 +247,18 @@ drw_rect(Drw *drw, int x, int y, unsigne
 }
 
 int
-drw_text(Drw *drw, int x, int y, unsigned int w, unsigned int h, unsigned int lpad, const char *text, int invert)
+drw_text(Drw* drw, int x, int y, unsigned int w, unsigned int h, unsigned int lpad, const char* text, int invert)
 {
 	int ty, ellipsis_x = 0;
 	unsigned int tmpw, ew, ellipsis_w = 0, ellipsis_len, hash, h0, h1;
-	XftDraw *d = NULL;
+	XftDraw* d = NULL;
 	Fnt *usedfont, *curfont, *nextfont;
 	int utf8strlen, utf8charlen, utf8err, render = x || y || w || h;
 	long utf8codepoint = 0;
-	const char *utf8str;
-	FcCharSet *fccharset;
-	FcPattern *fcpattern;
-	FcPattern *match;
+	const char* utf8str;
+	FcCharSet* fccharset;
+	FcPattern* fcpattern;
+	FcPattern* match;
 	XftResult result;
 	int charexists = 0, overflow = 0;
 	/* keep track of a couple codepoints for which we have no match. */
@@ -287,8 +289,8 @@ drw_text(Drw *drw, int x, int y, unsigne
 		invalid_width = drw_fontset_getwidth(drw, invalid);
 	while (1) {
 		ew = ellipsis_len = utf8err = utf8charlen = utf8strlen = 0;
-		utf8str = text;
-		nextfont = NULL;
+		utf8str                                                = text;
+		nextfont                                               = NULL;
 		while (*text) {
 			utf8charlen = utf8decode(text, &utf8codepoint, &utf8err);
 			for (curfont = drw->fonts; curfont; curfont = curfont->next) {
@@ -297,8 +299,8 @@ drw_text(Drw *drw, int x, int y, unsigne
 					drw_font_getexts(curfont, text, utf8charlen, &tmpw, NULL);
 					if (ew + ellipsis_width <= w) {
 						/* keep track where the ellipsis still fits */
-						ellipsis_x = x + ew;
-						ellipsis_w = w - ew;
+						ellipsis_x   = x + ew;
+						ellipsis_w   = w - ew;
 						ellipsis_len = utf8strlen;
 					}
 
@@ -332,7 +334,7 @@ drw_text(Drw *drw, int x, int y, unsigne
 			if (render) {
 				ty = y + (h - usedfont->h) / 2 + usedfont->xfont->ascent;
 				XftDrawStringUtf8(d, &drw->scheme[invert ? ColBg : ColFg],
-				                  usedfont->xfont, x, ty, (XftChar8 *)utf8str, utf8strlen);
+				                  usedfont->xfont, x, ty, (XftChar8*) utf8str, utf8strlen);
 			}
 			x += ew;
 			w -= ew;
@@ -350,17 +352,17 @@ drw_text(Drw *drw, int x, int y, unsigne
 			break;
 		} else if (nextfont) {
 			charexists = 0;
-			usedfont = nextfont;
+			usedfont   = nextfont;
 		} else {
 			/* Regardless of whether or not a fallback font is found, the
 			 * character must be drawn. */
 			charexists = 1;
 
-			hash = (unsigned int)utf8codepoint;
+			hash = (unsigned int) utf8codepoint;
 			hash = ((hash >> 16) ^ hash) * 0x21F0AAAD;
 			hash = ((hash >> 15) ^ hash) * 0xD35A2D97;
-			h0 = ((hash >> 15) ^ hash) % LENGTH(nomatches);
-			h1 = (hash >> 17) % LENGTH(nomatches);
+			h0   = ((hash >> 15) ^ hash) % LENGTH(nomatches);
+			h1   = (hash >> 17) % LENGTH(nomatches);
 			/* avoid expensive XftFontMatch call when we know we won't find a match */
 			if (nomatches[h0] == utf8codepoint || nomatches[h1] == utf8codepoint)
 				goto no_match;
@@ -406,7 +408,7 @@ no_match:
 }
 
 void
-drw_map(Drw *drw, Window win, int x, int y, unsigned int w, unsigned int h)
+drw_map(Drw* drw, Window win, int x, int y, unsigned int w, unsigned int h)
 {
 	if (!drw)
 		return;
@@ -416,7 +418,7 @@ drw_map(Drw *drw, Window win, int x, int
 }
 
 unsigned int
-drw_fontset_getwidth(Drw *drw, const char *text)
+drw_fontset_getwidth(Drw* drw, const char* text)
 {
 	if (!drw || !drw->fonts || !text)
 		return 0;
@@ -424,7 +426,7 @@ drw_fontset_getwidth(Drw *drw, const cha
 }
 
 unsigned int
-drw_fontset_getwidth_clamp(Drw *drw, const char *text, unsigned int n)
+drw_fontset_getwidth_clamp(Drw* drw, const char* text, unsigned int n)
 {
 	unsigned int tmp = 0;
 	if (drw && drw->fonts && text && n)
@@ -433,24 +435,24 @@ drw_fontset_getwidth_clamp(Drw *drw, con
 }
 
 void
-drw_font_getexts(Fnt *font, const char *text, unsigned int len, unsigned int *w, unsigned int *h)
+drw_font_getexts(Fnt* font, const char* text, unsigned int len, unsigned int* w, unsigned int* h)
 {
 	XGlyphInfo ext;
 
 	if (!font || !text)
 		return;
 
-	XftTextExtentsUtf8(font->dpy, font->xfont, (XftChar8 *)text, len, &ext);
+	XftTextExtentsUtf8(font->dpy, font->xfont, (XftChar8*) text, len, &ext);
 	if (w)
 		*w = ext.xOff;
 	if (h)
 		*h = font->h;
 }
 
-Cur *
-drw_cur_create(Drw *drw, int shape)
+Cur*
+drw_cur_create(Drw* drw, int shape)
 {
-	Cur *cur;
+	Cur* cur;
 
 	if (!drw || !(cur = ecalloc(1, sizeof(Cur))))
 		return NULL;
@@ -461,7 +463,7 @@ drw_cur_create(Drw *drw, int shape)
 }
 
 void
-drw_cur_free(Drw *drw, Cur *cursor)
+drw_cur_free(Drw* drw, Cur* cursor)
 {
 	if (!cursor)
 		return;
--- a/drw.h
+++ b/drw.h
@@ -5,56 +5,61 @@ typedef struct {
 } Cur;
 
 typedef struct Fnt {
-	Display *dpy;
+	Display* dpy;
 	unsigned int h;
-	XftFont *xfont;
-	FcPattern *pattern;
-	struct Fnt *next;
+	XftFont* xfont;
+	FcPattern* pattern;
+	struct Fnt* next;
 } Fnt;
 
-enum { ColFg, ColBg, ColBorder }; /* Clr scheme index */
+enum
+{
+	ColFg,
+	ColBg,
+	ColBorder,
+}; /* Clr scheme index */
 typedef XftColor Clr;
 
 typedef struct {
 	unsigned int w, h;
-	Display *dpy;
+	Display* dpy;
 	int screen;
 	Window root;
 	Drawable drawable;
 	GC gc;
-	Clr *scheme;
-	Fnt *fonts;
+	Clr* scheme;
+	Fnt* fonts;
 } Drw;
 
 /* Drawable abstraction */
-Drw *drw_create(Display *dpy, int screen, Window win, unsigned int w, unsigned int h);
-void drw_resize(Drw *drw, unsigned int w, unsigned int h);
-void drw_free(Drw *drw);
+Drw* drw_create(Display* dpy, int screen, Window win, unsigned int w, unsigned int h);
+void drw_resize(Drw* drw, unsigned int w, unsigned int h);
+void drw_free(Drw* drw);
 
 /* Fnt abstraction */
-Fnt *drw_fontset_create(Drw* drw, const char *fonts[], size_t fontcount);
+Fnt* drw_fontset_create(Drw* drw, const char* fonts[], size_t fontcount);
 void drw_fontset_free(Fnt* set);
-unsigned int drw_fontset_getwidth(Drw *drw, const char *text);
-unsigned int drw_fontset_getwidth_clamp(Drw *drw, const char *text, unsigned int n);
-void drw_font_getexts(Fnt *font, const char *text, unsigned int len, unsigned int *w, unsigned int *h);
+unsigned int drw_fontset_getwidth(Drw* drw, const char* text);
+unsigned int drw_fontset_getwidth_clamp(Drw* drw, const char* text, unsigned int n);
+void drw_font_getexts(Fnt* font, const char* text, unsigned int len, unsigned int* w, unsigned int* h);
 
 /* Colorscheme abstraction */
-void drw_clr_create(Drw *drw, Clr *dest, const char *clrname);
-void drw_clr_free(Drw *drw, Clr *c);
-Clr *drw_scm_create(Drw *drw, const char *clrnames[], size_t clrcount);
-void drw_scm_free(Drw *drw, Clr *scm, size_t clrcount);
+void drw_clr_create(Drw* drw, Clr* dest, const char* clrname);
+void drw_clr_free(Drw* drw, Clr* c);
+Clr* drw_scm_create(Drw* drw, const char* clrnames[], size_t clrcount);
+void drw_scm_free(Drw* drw, Clr* scm, size_t clrcount);
 
 /* Cursor abstraction */
-Cur *drw_cur_create(Drw *drw, int shape);
-void drw_cur_free(Drw *drw, Cur *cursor);
+Cur* drw_cur_create(Drw* drw, int shape);
+void drw_cur_free(Drw* drw, Cur* cursor);
 
 /* Drawing context manipulation */
-void drw_setfontset(Drw *drw, Fnt *set);
-void drw_setscheme(Drw *drw, Clr *scm);
+void drw_setfontset(Drw* drw, Fnt* set);
+void drw_setscheme(Drw* drw, Clr* scm);
 
 /* Drawing functions */
-void drw_rect(Drw *drw, int x, int y, unsigned int w, unsigned int h, int filled, int invert);
-int drw_text(Drw *drw, int x, int y, unsigned int w, unsigned int h, unsigned int lpad, const char *text, int invert);
+void drw_rect(Drw* drw, int x, int y, unsigned int w, unsigned int h, int filled, int invert);
+int drw_text(Drw* drw, int x, int y, unsigned int w, unsigned int h, unsigned int lpad, const char* text, int invert);
 
 /* Map functions */
-void drw_map(Drw *drw, Window win, int x, int y, unsigned int w, unsigned int h);
+void drw_map(Drw* drw, Window win, int x, int y, unsigned int w, unsigned int h);
--- a/dwm.c
+++ b/dwm.c
@@ -27,15 +27,15 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <unistd.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <X11/cursorfont.h>
-#include <X11/keysym.h>
+#include <unistd.h>
 #include <X11/Xatom.h>
 #include <X11/Xlib.h>
 #include <X11/Xproto.h>
 #include <X11/Xutil.h>
+#include <X11/cursorfont.h>
+#include <X11/keysym.h>
 #ifdef XINERAMA
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
@@ -47,7 +47,7 @@
 /* macros */
 #define BUTTONMASK              (ButtonPressMask|ButtonReleaseMask)
 #define CLEANMASK(mask)         (mask & ~(numlockmask|LockMask) & (ShiftMask|ControlMask|Mod1Mask|Mod2Mask|Mod3Mask|Mod4Mask|Mod5Mask))
-#define INTERSECT(x,y,w,h,m)    (MAX(0, MIN((x)+(w),(m)->wx+(m)->ww) - MAX((x),(m)->wx)) \
+#define INTERSECT(x, y, w, h, m)    (MAX(0, MIN((x)+(w),(m)->wx+(m)->ww) - MAX((x),(m)->wx)) \
                                * MAX(0, MIN((y)+(h),(m)->wy+(m)->wh) - MAX((y),(m)->wy)))
 #define ISVISIBLE(C)            ((C->tags & C->mon->tagset[C->mon->seltags]))
 #define MOUSEMASK               (BUTTONMASK|PointerMotionMask)
@@ -57,27 +57,62 @@
 #define TEXTW(X)                (drw_fontset_getwidth(drw, (X)) + lrpad)
 
 /* enums */
-enum { CurNormal, CurResize, CurMove, CurLast }; /* cursor */
-enum { SchemeNorm, SchemeSel }; /* color schemes */
-enum { NetSupported, NetWMName, NetWMState, NetWMCheck,
-       NetWMFullscreen, NetActiveWindow, NetWMWindowType,
-       NetWMWindowTypeDialog, NetClientList, NetLast }; /* EWMH atoms */
-enum { WMProtocols, WMDelete, WMState, WMTakeFocus, WMLast }; /* default atoms */
-enum { ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle,
-       ClkClientWin, ClkRootWin, ClkLast }; /* clicks */
+enum
+{
+	CurNormal,
+	CurResize,
+	CurMove,
+	CurLast,
+}; /* cursor */
+enum
+{
+	SchemeNorm,
+	SchemeSel,
+}; /* color schemes */
+enum
+{
+	NetSupported,
+	NetWMName,
+	NetWMState,
+	NetWMCheck,
+	NetWMFullscreen,
+	NetActiveWindow,
+	NetWMWindowType,
+	NetWMWindowTypeDialog,
+	NetClientList,
+	NetLast,
+}; /* EWMH atoms */
+enum
+{
+	WMProtocols,
+	WMDelete,
+	WMState,
+	WMTakeFocus,
+	WMLast,
+}; /* default atoms */
+enum
+{
+	ClkTagBar,
+	ClkLtSymbol,
+	ClkStatusText,
+	ClkWinTitle,
+	ClkClientWin,
+	ClkRootWin,
+	ClkLast,
+}; /* clicks */
 
 typedef union {
 	int i;
 	unsigned int ui;
 	float f;
-	const void *v;
+	const void* v;
 } Arg;
 
 typedef struct {
 	unsigned int click;
 	unsigned int mask;
 	unsigned int button;
-	void (*func)(const Arg *arg);
+	void (*func)(const Arg* arg);
 	const Arg arg;
 } Button;
 
@@ -92,22 +127,22 @@ struct Client {
 	int bw, oldbw;
 	unsigned int tags;
 	int isfixed, isfloating, isurgent, neverfocus, oldstate, isfullscreen;
-	Client *next;
-	Client *snext;
-	Monitor *mon;
+	Client* next;
+	Client* snext;
+	Monitor* mon;
 	Window win;
 };
 
 typedef struct {
 	unsigned int mod;
 	KeySym keysym;
-	void (*func)(const Arg *);
+	void (*func)(const Arg*);
 	const Arg arg;
 } Key;
 
 typedef struct {
-	const char *symbol;
-	void (*arrange)(Monitor *);
+	const char* symbol;
+	void (*arrange)(Monitor*);
 } Layout;
 
 struct Monitor {
@@ -115,155 +150,155 @@ struct Monitor {
 	float mfact;
 	int nmaster;
 	int num;
-	int by;               /* bar geometry */
-	int mx, my, mw, mh;   /* screen size */
-	int wx, wy, ww, wh;   /* window area  */
+	int by;             /* bar geometry */
+	int mx, my, mw, mh; /* screen size */
+	int wx, wy, ww, wh; /* window area  */
 	unsigned int seltags;
 	unsigned int sellt;
 	unsigned int tagset[2];
 	int showbar;
 	int topbar;
-	Client *clients;
-	Client *sel;
-	Client *stack;
-	Monitor *next;
+	Client* clients;
+	Client* sel;
+	Client* stack;
+	Monitor* next;
 	Window barwin;
-	const Layout *lt[2];
+	const Layout* lt[2];
 };
 
 typedef struct {
-	const char *class;
-	const char *instance;
-	const char *title;
+	const char* class;
+	const char* instance;
+	const char* title;
 	unsigned int tags;
 	int isfloating;
 	int monitor;
 } Rule;
 
 /* function declarations */
-static void applyrules(Client *c);
-static int applysizehints(Client *c, int *x, int *y, int *w, int *h, int interact);
-static void arrange(Monitor *m);
-static void arrangemon(Monitor *m);
-static void attach(Client *c);
-static void attachstack(Client *c);
-static void buttonpress(XEvent *e);
+static void applyrules(Client* c);
+static int applysizehints(Client* c, int* x, int* y, int* w, int* h, int interact);
+static void arrange(Monitor* m);
+static void arrangemon(Monitor* m);
+static void attach(Client* c);
+static void attachstack(Client* c);
+static void buttonpress(XEvent* e);
 static void checkotherwm(void);
 static void cleanup(void);
-static void cleanupmon(Monitor *mon);
-static void clientmessage(XEvent *e);
-static void configure(Client *c);
-static void configurenotify(XEvent *e);
-static void configurerequest(XEvent *e);
-static Monitor *createmon(void);
-static void destroynotify(XEvent *e);
-static void detach(Client *c);
-static void detachstack(Client *c);
-static Monitor *dirtomon(int dir);
-static void drawbar(Monitor *m);
+static void cleanupmon(Monitor* mon);
+static void clientmessage(XEvent* e);
+static void configure(Client* c);
+static void configurenotify(XEvent* e);
+static void configurerequest(XEvent* e);
+static Monitor* createmon(void);
+static void destroynotify(XEvent* e);
+static void detach(Client* c);
+static void detachstack(Client* c);
+static Monitor* dirtomon(int dir);
+static void drawbar(Monitor* m);
 static void drawbars(void);
-static void enternotify(XEvent *e);
-static void expose(XEvent *e);
-static void focus(Client *c);
-static void focusin(XEvent *e);
-static void focusmon(const Arg *arg);
-static void focusstack(const Arg *arg);
-static Atom getatomprop(Client *c, Atom prop);
-static int getrootptr(int *x, int *y);
+static void enternotify(XEvent* e);
+static void expose(XEvent* e);
+static void focus(Client* c);
+static void focusin(XEvent* e);
+static void focusmon(const Arg* arg);
+static void focusstack(const Arg* arg);
+static Atom getatomprop(Client* c, Atom prop);
+static int getrootptr(int* x, int* y);
 static long getstate(Window w);
-static int gettextprop(Window w, Atom atom, char *text, unsigned int size);
-static void grabbuttons(Client *c, int focused);
+static int gettextprop(Window w, Atom atom, char* text, unsigned int size);
+static void grabbuttons(Client* c, int focused);
 static void grabkeys(void);
-static void incnmaster(const Arg *arg);
-static void keypress(XEvent *e);
-static void killclient(const Arg *arg);
-static void manage(Window w, XWindowAttributes *wa);
-static void mappingnotify(XEvent *e);
-static void maprequest(XEvent *e);
-static void monocle(Monitor *m);
-static void motionnotify(XEvent *e);
-static void movemouse(const Arg *arg);
-static Client *nexttiled(Client *c);
-static void pop(Client *c);
-static void propertynotify(XEvent *e);
-static void quit(const Arg *arg);
-static Monitor *recttomon(int x, int y, int w, int h);
-static void resize(Client *c, int x, int y, int w, int h, int interact);
-static void resizeclient(Client *c, int x, int y, int w, int h);
-static void resizemouse(const Arg *arg);
-static void restack(Monitor *m);
+static void incnmaster(const Arg* arg);
+static void keypress(XEvent* e);
+static void killclient(const Arg* arg);
+static void manage(Window w, XWindowAttributes* wa);
+static void mappingnotify(XEvent* e);
+static void maprequest(XEvent* e);
+static void monocle(Monitor* m);
+static void motionnotify(XEvent* e);
+static void movemouse(const Arg* arg);
+static Client* nexttiled(Client* c);
+static void pop(Client* c);
+static void propertynotify(XEvent* e);
+static void quit(const Arg* arg);
+static Monitor* recttomon(int x, int y, int w, int h);
+static void resize(Client* c, int x, int y, int w, int h, int interact);
+static void resizeclient(Client* c, int x, int y, int w, int h);
+static void resizemouse(const Arg* arg);
+static void restack(Monitor* m);
 static void run(void);
 static void scan(void);
-static int sendevent(Client *c, Atom proto);
-static void sendmon(Client *c, Monitor *m);
-static void setclientstate(Client *c, long state);
-static void setfocus(Client *c);
-static void setfullscreen(Client *c, int fullscreen);
-static void setlayout(const Arg *arg);
-static void setmfact(const Arg *arg);
+static int sendevent(Client* c, Atom proto);
+static void sendmon(Client* c, Monitor* m);
+static void setclientstate(Client* c, long state);
+static void setfocus(Client* c);
+static void setfullscreen(Client* c, int fullscreen);
+static void setlayout(const Arg* arg);
+static void setmfact(const Arg* arg);
 static void setup(void);
-static void seturgent(Client *c, int urg);
-static void showhide(Client *c);
-static void spawn(const Arg *arg);
-static void tag(const Arg *arg);
-static void tagmon(const Arg *arg);
-static void tile(Monitor *m);
-static void togglebar(const Arg *arg);
-static void togglefloating(const Arg *arg);
-static void toggletag(const Arg *arg);
-static void toggleview(const Arg *arg);
-static void unfocus(Client *c, int setfocus);
-static void unmanage(Client *c, int destroyed);
-static void unmapnotify(XEvent *e);
-static void updatebarpos(Monitor *m);
+static void seturgent(Client* c, int urg);
+static void showhide(Client* c);
+static void spawn(const Arg* arg);
+static void tag(const Arg* arg);
+static void tagmon(const Arg* arg);
+static void tile(Monitor* m);
+static void togglebar(const Arg* arg);
+static void togglefloating(const Arg* arg);
+static void toggletag(const Arg* arg);
+static void toggleview(const Arg* arg);
+static void unfocus(Client* c, int setfocus);
+static void unmanage(Client* c, int destroyed);
+static void unmapnotify(XEvent* e);
+static void updatebarpos(Monitor* m);
 static void updatebars(void);
 static void updateclientlist(void);
 static int updategeom(void);
 static void updatenumlockmask(void);
-static void updatesizehints(Client *c);
+static void updatesizehints(Client* c);
 static void updatestatus(void);
-static void updatetitle(Client *c);
-static void updatewindowtype(Client *c);
-static void updatewmhints(Client *c);
-static void view(const Arg *arg);
-static Client *wintoclient(Window w);
-static Monitor *wintomon(Window w);
-static int xerror(Display *dpy, XErrorEvent *ee);
-static int xerrordummy(Display *dpy, XErrorEvent *ee);
-static int xerrorstart(Display *dpy, XErrorEvent *ee);
-static void zoom(const Arg *arg);
+static void updatetitle(Client* c);
+static void updatewindowtype(Client* c);
+static void updatewmhints(Client* c);
+static void view(const Arg* arg);
+static Client* wintoclient(Window w);
+static Monitor* wintomon(Window w);
+static int xerror(Display* dpy, XErrorEvent* ee);
+static int xerrordummy(Display* dpy, XErrorEvent* ee);
+static int xerrorstart(Display* dpy, XErrorEvent* ee);
+static void zoom(const Arg* arg);
 
 /* variables */
 static const char broken[] = "broken";
 static char stext[256];
 static int screen;
-static int sw, sh;           /* X display screen geometry width, height */
-static int bh;               /* bar height */
-static int lrpad;            /* sum of left and right padding for text */
-static int (*xerrorxlib)(Display *, XErrorEvent *);
-static unsigned int numlockmask = 0;
-static void (*handler[LASTEvent]) (XEvent *) = {
-	[ButtonPress] = buttonpress,
-	[ClientMessage] = clientmessage,
+static int sw, sh; /* X display screen geometry width, height */
+static int bh;     /* bar height */
+static int lrpad;  /* sum of left and right padding for text */
+static int (*xerrorxlib)(Display*, XErrorEvent*);
+static unsigned int numlockmask            = 0;
+static void (*handler[LASTEvent])(XEvent*) = {
+	[ButtonPress]      = buttonpress,
+	[ClientMessage]    = clientmessage,
 	[ConfigureRequest] = configurerequest,
-	[ConfigureNotify] = configurenotify,
-	[DestroyNotify] = destroynotify,
-	[EnterNotify] = enternotify,
-	[Expose] = expose,
-	[FocusIn] = focusin,
-	[KeyPress] = keypress,
-	[MappingNotify] = mappingnotify,
-	[MapRequest] = maprequest,
-	[MotionNotify] = motionnotify,
-	[PropertyNotify] = propertynotify,
-	[UnmapNotify] = unmapnotify
+	[ConfigureNotify]  = configurenotify,
+	[DestroyNotify]    = destroynotify,
+	[EnterNotify]      = enternotify,
+	[Expose]           = expose,
+	[FocusIn]          = focusin,
+	[KeyPress]         = keypress,
+	[MappingNotify]    = mappingnotify,
+	[MapRequest]       = maprequest,
+	[MotionNotify]     = motionnotify,
+	[PropertyNotify]   = propertynotify,
+	[UnmapNotify]      = unmapnotify
 };
 static Atom wmatom[WMLast], netatom[NetLast];
 static int running = 1;
-static Cur *cursor[CurLast];
-static Clr **scheme;
-static Display *dpy;
-static Drw *drw;
+static Cur* cursor[CurLast];
+static Clr** scheme;
+static Display* dpy;
+static Drw* drw;
 static Monitor *mons, *selmon;
 static Window root, wmcheckwin;
 
@@ -271,34 +306,34 @@ static Window root, wmcheckwin;
 #include "config.h"
 
 /* compile-time check if all tags fit into an unsigned int bit array. */
-struct NumTags { char limitexceeded[LENGTH(tags) > 31 ? -1 : 1]; };
+struct NumTags {
+	char limitexceeded[LENGTH(tags) > 31 ? -1 : 1];
+};
 
 /* function implementations */
 void
-applyrules(Client *c)
+applyrules(Client* c)
 {
 	const char *class, *instance;
 	unsigned int i;
-	const Rule *r;
-	Monitor *m;
+	const Rule* r;
+	Monitor* m;
 	XClassHint ch = { NULL, NULL };
 
 	/* rule matching */
 	c->isfloating = 0;
-	c->tags = 0;
+	c->tags       = 0;
 	XGetClassHint(dpy, c->win, &ch);
 	class    = ch.res_class ? ch.res_class : broken;
-	instance = ch.res_name  ? ch.res_name  : broken;
+	instance = ch.res_name ? ch.res_name : broken;
 
 	for (i = 0; i < LENGTH(rules); i++) {
 		r = &rules[i];
-		if ((!r->title || strstr(c->name, r->title))
-		&& (!r->class || strstr(class, r->class))
-		&& (!r->instance || strstr(instance, r->instance)))
-		{
+		if ((!r->title || strstr(c->name, r->title)) && (!r->class || strstr(class, r->class)) && (!r->instance || strstr(instance, r->instance))) {
 			c->isfloating = r->isfloating;
 			c->tags |= r->tags;
-			for (m = mons; m && m->num != r->monitor; m = m->next);
+			for (m = mons; m && m->num != r->monitor; m = m->next)
+				;
 			if (m)
 				c->mon = m;
 		}
@@ -311,10 +346,10 @@ applyrules(Client *c)
 }
 
 int
-applysizehints(Client *c, int *x, int *y, int *w, int *h, int interact)
+applysizehints(Client* c, int* x, int* y, int* w, int* h, int interact)
 {
 	int baseismin;
-	Monitor *m = c->mon;
+	Monitor* m = c->mon;
 
 	/* set minimum possible */
 	*w = MAX(1, *w);
@@ -353,9 +388,9 @@ applysizehints(Client *c, int *x, int *y
 		}
 		/* adjust for aspect limits */
 		if (c->mina > 0 && c->maxa > 0) {
-			if (c->maxa < (float)*w / *h)
+			if (c->maxa < (float) *w / *h)
 				*w = *h * c->maxa + 0.5;
-			else if (c->mina < (float)*h / *w)
+			else if (c->mina < (float) *h / *w)
 				*h = *w * c->mina + 0.5;
 		}
 		if (baseismin) { /* increment calculation requires this */
@@ -379,21 +414,23 @@ applysizehints(Client *c, int *x, int *y
 }
 
 void
-arrange(Monitor *m)
+arrange(Monitor* m)
 {
 	if (m)
 		showhide(m->stack);
-	else for (m = mons; m; m = m->next)
-		showhide(m->stack);
+	else
+		for (m = mons; m; m = m->next)
+			showhide(m->stack);
 	if (m) {
 		arrangemon(m);
 		restack(m);
-	} else for (m = mons; m; m = m->next)
-		arrangemon(m);
+	} else
+		for (m = mons; m; m = m->next)
+			arrangemon(m);
 }
 
 void
-arrangemon(Monitor *m)
+arrangemon(Monitor* m)
 {
 	strncpy(m->ltsymbol, m->lt[m->sellt]->symbol, sizeof m->ltsymbol);
 	if (m->lt[m->sellt]->arrange)
@@ -401,27 +438,27 @@ arrangemon(Monitor *m)
 }
 
 void
-attach(Client *c)
+attach(Client* c)
 {
-	c->next = c->mon->clients;
+	c->next         = c->mon->clients;
 	c->mon->clients = c;
 }
 
 void
-attachstack(Client *c)
+attachstack(Client* c)
 {
-	c->snext = c->mon->stack;
+	c->snext      = c->mon->stack;
 	c->mon->stack = c;
 }
 
 void
-buttonpress(XEvent *e)
+buttonpress(XEvent* e)
 {
 	unsigned int i, x, click;
-	Arg arg = {0};
-	Client *c;
-	Monitor *m;
-	XButtonPressedEvent *ev = &e->xbutton;
+	Arg arg = { 0 };
+	Client* c;
+	Monitor* m;
+	XButtonPressedEvent* ev = &e->xbutton;
 
 	click = ClkRootWin;
 	/* focus monitor if necessary */
@@ -436,11 +473,11 @@ buttonpress(XEvent *e)
 			x += TEXTW(tags[i]);
 		while (ev->x >= x && ++i < LENGTH(tags));
 		if (i < LENGTH(tags)) {
-			click = ClkTagBar;
+			click  = ClkTagBar;
 			arg.ui = 1 << i;
 		} else if (ev->x < x + TEXTW(selmon->ltsymbol))
 			click = ClkLtSymbol;
-		else if (ev->x > selmon->ww - (int)TEXTW(stext) + lrpad - 2)
+		else if (ev->x > selmon->ww - (int) TEXTW(stext) + lrpad - 2)
 			click = ClkStatusText;
 		else
 			click = ClkWinTitle;
@@ -451,8 +488,7 @@ buttonpress(XEvent *e)
 		click = ClkClientWin;
 	}
 	for (i = 0; i < LENGTH(buttons); i++)
-		if (click == buttons[i].click && buttons[i].func && buttons[i].button == ev->button
-		&& CLEANMASK(buttons[i].mask) == CLEANMASK(ev->state))
+		if (click == buttons[i].click && buttons[i].func && buttons[i].button == ev->button && CLEANMASK(buttons[i].mask) == CLEANMASK(ev->state))
 			buttons[i].func(click == ClkTagBar && buttons[i].arg.i == 0 ? &arg : &buttons[i].arg);
 }
 
@@ -470,9 +506,9 @@ checkotherwm(void)
 void
 cleanup(void)
 {
-	Arg a = {.ui = ~0};
+	Arg a      = { .ui = ~0 };
 	Layout foo = { "", NULL };
-	Monitor *m;
+	Monitor* m;
 	size_t i;
 
 	view(&a);
@@ -496,14 +532,15 @@ cleanup(void)
 }
 
 void
-cleanupmon(Monitor *mon)
+cleanupmon(Monitor* mon)
 {
-	Monitor *m;
+	Monitor* m;
 
 	if (mon == mons)
 		mons = mons->next;
 	else {
-		for (m = mons; m && m->next != mon; m = m->next);
+		for (m = mons; m && m->next != mon; m = m->next)
+			;
 		m->next = mon->next;
 	}
 	XUnmapWindow(dpy, mon->barwin);
@@ -512,18 +549,17 @@ cleanupmon(Monitor *mon)
 }
 
 void
-clientmessage(XEvent *e)
+clientmessage(XEvent* e)
 {
-	XClientMessageEvent *cme = &e->xclient;
-	Client *c = wintoclient(cme->window);
+	XClientMessageEvent* cme = &e->xclient;
+	Client* c                = wintoclient(cme->window);
 
 	if (!c)
 		return;
 	if (cme->message_type == netatom[NetWMState]) {
-		if (cme->data.l[1] == netatom[NetWMFullscreen]
-		|| cme->data.l[2] == netatom[NetWMFullscreen])
+		if (cme->data.l[1] == netatom[NetWMFullscreen] || cme->data.l[2] == netatom[NetWMFullscreen])
 			setfullscreen(c, (cme->data.l[0] == 1 /* _NET_WM_STATE_ADD    */
-				|| (cme->data.l[0] == 2 /* _NET_WM_STATE_TOGGLE */ && !c->isfullscreen)));
+			                  || (cme->data.l[0] == 2 /* _NET_WM_STATE_TOGGLE */ && !c->isfullscreen)));
 	} else if (cme->message_type == netatom[NetActiveWindow]) {
 		if (c != selmon->sel && !c->isurgent)
 			seturgent(c, 1);
@@ -531,37 +567,37 @@ clientmessage(XEvent *e)
 }
 
 void
-configure(Client *c)
+configure(Client* c)
 {
 	XConfigureEvent ce;
 
-	ce.type = ConfigureNotify;
-	ce.display = dpy;
-	ce.event = c->win;
-	ce.window = c->win;
-	ce.x = c->x;
-	ce.y = c->y;
-	ce.width = c->w;
-	ce.height = c->h;
-	ce.border_width = c->bw;
-	ce.above = None;
+	ce.type              = ConfigureNotify;
+	ce.display           = dpy;
+	ce.event             = c->win;
+	ce.window            = c->win;
+	ce.x                 = c->x;
+	ce.y                 = c->y;
+	ce.width             = c->w;
+	ce.height            = c->h;
+	ce.border_width      = c->bw;
+	ce.above             = None;
 	ce.override_redirect = False;
-	XSendEvent(dpy, c->win, False, StructureNotifyMask, (XEvent *)&ce);
+	XSendEvent(dpy, c->win, False, StructureNotifyMask, (XEvent*) &ce);
 }
 
 void
-configurenotify(XEvent *e)
+configurenotify(XEvent* e)
 {
-	Monitor *m;
-	Client *c;
-	XConfigureEvent *ev = &e->xconfigure;
+	Monitor* m;
+	Client* c;
+	XConfigureEvent* ev = &e->xconfigure;
 	int dirty;
 
 	/* TODO: updategeom handling sucks, needs to be simplified */
 	if (ev->window == root) {
 		dirty = (sw != ev->width || sh != ev->height);
-		sw = ev->width;
-		sh = ev->height;
+		sw    = ev->width;
+		sh    = ev->height;
 		if (updategeom() || dirty) {
 			drw_resize(drw, sw, bh);
 			updatebars();
@@ -578,11 +614,11 @@ configurenotify(XEvent *e)
 }
 
 void
-configurerequest(XEvent *e)
+configurerequest(XEvent* e)
 {
-	Client *c;
-	Monitor *m;
-	XConfigureRequestEvent *ev = &e->xconfigurerequest;
+	Client* c;
+	Monitor* m;
+	XConfigureRequestEvent* ev = &e->xconfigurerequest;
 	XWindowChanges wc;
 
 	if ((c = wintoclient(ev->window))) {
@@ -592,122 +628,127 @@ configurerequest(XEvent *e)
 			m = c->mon;
 			if (ev->value_mask & CWX) {
 				c->oldx = c->x;
-				c->x = m->mx + ev->x;
+				c->x    = m->mx + ev->x;
 			}
 			if (ev->value_mask & CWY) {
 				c->oldy = c->y;
-				c->y = m->my + ev->y;
+				c->y    = m->my + ev->y;
 			}
 			if (ev->value_mask & CWWidth) {
 				c->oldw = c->w;
-				c->w = ev->width;
+				c->w    = ev->width;
 			}
 			if (ev->value_mask & CWHeight) {
 				c->oldh = c->h;
-				c->h = ev->height;
+				c->h    = ev->height;
 			}
 			if ((c->x + c->w) > m->mx + m->mw && c->isfloating)
-				c->x = m->mx + (m->mw / 2 - WIDTH(c) / 2); /* center in x direction */
+				c->x = m->mx + (m->mw / 2 - WIDTH(c) / 2);  /* center in x direction */
 			if ((c->y + c->h) > m->my + m->mh && c->isfloating)
 				c->y = m->my + (m->mh / 2 - HEIGHT(c) / 2); /* center in y direction */
-			if ((ev->value_mask & (CWX|CWY)) && !(ev->value_mask & (CWWidth|CWHeight)))
+			if ((ev->value_mask & (CWX | CWY)) && !(ev->value_mask & (CWWidth | CWHeight)))
 				configure(c);
 			if (ISVISIBLE(c))
 				XMoveResizeWindow(dpy, c->win, c->x, c->y, c->w, c->h);
 		} else
 			configure(c);
 	} else {
-		wc.x = ev->x;
-		wc.y = ev->y;
-		wc.width = ev->width;
-		wc.height = ev->height;
+		wc.x            = ev->x;
+		wc.y            = ev->y;
+		wc.width        = ev->width;
+		wc.height       = ev->height;
 		wc.border_width = ev->border_width;
-		wc.sibling = ev->above;
-		wc.stack_mode = ev->detail;
+		wc.sibling      = ev->above;
+		wc.stack_mode   = ev->detail;
 		XConfigureWindow(dpy, ev->window, ev->value_mask, &wc);
 	}
 	XSync(dpy, False);
 }
 
-Monitor *
+Monitor*
 createmon(void)
 {
-	Monitor *m;
+	Monitor* m;
 
-	m = ecalloc(1, sizeof(Monitor));
+	m            = ecalloc(1, sizeof(Monitor));
 	m->tagset[0] = m->tagset[1] = 1;
-	m->mfact = mfact;
-	m->nmaster = nmaster;
-	m->showbar = showbar;
-	m->topbar = topbar;
-	m->lt[0] = &layouts[0];
-	m->lt[1] = &layouts[1 % LENGTH(layouts)];
+	m->mfact                    = mfact;
+	m->nmaster                  = nmaster;
+	m->showbar                  = showbar;
+	m->topbar                   = topbar;
+	m->lt[0]                    = &layouts[0];
+	m->lt[1]                    = &layouts[1 % LENGTH(layouts)];
 	strncpy(m->ltsymbol, layouts[0].symbol, sizeof m->ltsymbol);
 	return m;
 }
 
 void
-destroynotify(XEvent *e)
+destroynotify(XEvent* e)
 {
-	Client *c;
-	XDestroyWindowEvent *ev = &e->xdestroywindow;
+	Client* c;
+	XDestroyWindowEvent* ev = &e->xdestroywindow;
 
 	if ((c = wintoclient(ev->window)))
 		unmanage(c, 1);
 }
 
 void
-detach(Client *c)
+detach(Client* c)
 {
-	Client **tc;
+	Client** tc;
 
-	for (tc = &c->mon->clients; *tc && *tc != c; tc = &(*tc)->next);
+	for (tc = &c->mon->clients; *tc && *tc != c; tc = &(*tc)->next)
+		;
 	*tc = c->next;
 }
 
 void
-detachstack(Client *c)
+detachstack(Client* c)
 {
 	Client **tc, *t;
 
-	for (tc = &c->mon->stack; *tc && *tc != c; tc = &(*tc)->snext);
+	for (tc = &c->mon->stack; *tc && *tc != c; tc = &(*tc)->snext)
+		;
 	*tc = c->snext;
 
 	if (c == c->mon->sel) {
-		for (t = c->mon->stack; t && !ISVISIBLE(t); t = t->snext);
+		for (t = c->mon->stack; t && !ISVISIBLE(t); t = t->snext)
+			;
 		c->mon->sel = t;
 	}
 }
 
-Monitor *
+Monitor*
 dirtomon(int dir)
 {
-	Monitor *m = NULL;
+	Monitor* m = NULL;
 
 	if (dir > 0) {
 		if (!(m = selmon->next))
 			m = mons;
 	} else if (selmon == mons)
-		for (m = mons; m->next; m = m->next);
+		for (m = mons; m->next; m = m->next)
+			;
 	else
-		for (m = mons; m->next != selmon; m = m->next);
+		for (m = mons; m->next != selmon; m = m->next)
+			;
 	return m;
 }
 
 void
-drawbar(Monitor *m)
+drawbar(Monitor* m)
 {
 	int x, w, tw = 0;
 	int boxs = drw->fonts->h / 9;
 	int boxw = drw->fonts->h / 6 + 2;
 	unsigned int i, occ = 0, urg = 0;
-	Client *c;
+	Client* c;
 
 	if (!m->showbar)
 		return;
 
 	/* draw status first so it can be overdrawn by tags later */
-	if (m == selmon) { /* status is only drawn on selected monitor */
+	if (m == selmon) {                 /* status is only drawn on selected monitor */
 		drw_setscheme(drw, scheme[SchemeNorm]);
 		tw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
 		drw_text(drw, m->ww - tw, 0, tw, bh, 0, stext, 0);
@@ -725,8 +766,8 @@ drawbar(Monitor *m)
 		drw_text(drw, x, 0, w, bh, lrpad / 2, tags[i], urg & 1 << i);
 		if (occ & 1 << i)
 			drw_rect(drw, x + boxs, boxs, boxw, boxw,
-				m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
-				urg & 1 << i);
+			         m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
+			         urg & 1 << i);
 		x += w;
 	}
 	w = TEXTW(m->ltsymbol);
@@ -750,18 +791,18 @@ drawbar(Monitor *m)
 void
 drawbars(void)
 {
-	Monitor *m;
+	Monitor* m;
 
 	for (m = mons; m; m = m->next)
 		drawbar(m);
 }
 
 void
-enternotify(XEvent *e)
+enternotify(XEvent* e)
 {
-	Client *c;
-	Monitor *m;
-	XCrossingEvent *ev = &e->xcrossing;
+	Client* c;
+	Monitor* m;
+	XCrossingEvent* ev = &e->xcrossing;
 
 	if ((ev->mode != NotifyNormal || ev->detail == NotifyInferior) && ev->window != root)
 		return;
@@ -776,20 +817,21 @@ enternotify(XEvent *e)
 }
 
 void
-expose(XEvent *e)
+expose(XEvent* e)
 {
-	Monitor *m;
-	XExposeEvent *ev = &e->xexpose;
+	Monitor* m;
+	XExposeEvent* ev = &e->xexpose;
 
 	if (ev->count == 0 && (m = wintomon(ev->window)))
 		drawbar(m);
 }
 
 void
-focus(Client *c)
+focus(Client* c)
 {
 	if (!c || !ISVISIBLE(c))
-		for (c = selmon->stack; c && !ISVISIBLE(c); c = c->snext);
+		for (c = selmon->stack; c && !ISVISIBLE(c); c = c->snext)
+			;
 	if (selmon->sel && selmon->sel != c)
 		unfocus(selmon->sel, 0);
 	if (c) {
@@ -812,18 +854,18 @@ focus(Client *c)
 
 /* there are some broken focus acquiring clients needing extra handling */
 void
-focusin(XEvent *e)
+focusin(XEvent* e)
 {
-	XFocusChangeEvent *ev = &e->xfocus;
+	XFocusChangeEvent* ev = &e->xfocus;
 
 	if (selmon->sel && ev->window != selmon->sel->win)
 		setfocus(selmon->sel);
 }
 
 void
-focusmon(const Arg *arg)
+focusmon(const Arg* arg)
 {
-	Monitor *m;
+	Monitor* m;
 
 	if (!mons->next)
 		return;
@@ -835,16 +877,18 @@ focusmon(const Arg *arg)
 }
 
 void
-focusstack(const Arg *arg)
+focusstack(const Arg* arg)
 {
 	Client *c = NULL, *i;
 
 	if (!selmon->sel || (selmon->sel->isfullscreen && lockfullscreen))
 		return;
 	if (arg->i > 0) {
-		for (c = selmon->sel->next; c && !ISVISIBLE(c); c = c->next);
+		for (c = selmon->sel->next; c && !ISVISIBLE(c); c = c->next)
+			;
 		if (!c)
-			for (c = selmon->clients; c && !ISVISIBLE(c); c = c->next);
+			for (c = selmon->clients; c && !ISVISIBLE(c); c = c->next)
+				;
 	} else {
 		for (i = selmon->clients; i != selmon->sel; i = i->next)
 			if (ISVISIBLE(i))
@@ -861,24 +905,25 @@ focusstack(const Arg *arg)
 }
 
 Atom
-getatomprop(Client *c, Atom prop)
+getatomprop(Client* c, Atom prop)
 {
 	int format;
 	unsigned long nitems, dl;
-	unsigned char *p = NULL;
+	unsigned char* p = NULL;
 	Atom da, atom = None;
 
 	if (XGetWindowProperty(dpy, c->win, prop, 0L, sizeof atom, False, XA_ATOM,
-		&da, &format, &nitems, &dl, &p) == Success && p) {
+	                       &da, &format, &nitems, &dl, &p) == Success &&
+	    p) {
 		if (nitems > 0 && format == 32)
-			atom = *(long *)p;
+			atom = *(long*) p;
 		XFree(p);
 	}
 	return atom;
 }
 
 int
-getrootptr(int *x, int *y)
+getrootptr(int* x, int* y)
 {
 	int di;
 	unsigned int dui;
@@ -891,24 +936,24 @@ long
 getstate(Window w)
 {
 	int format;
-	long result = -1;
-	unsigned char *p = NULL;
+	long result      = -1;
+	unsigned char* p = NULL;
 	unsigned long n, extra;
 	Atom real;
 
 	if (XGetWindowProperty(dpy, w, wmatom[WMState], 0L, 2L, False, wmatom[WMState],
-		&real, &format, &n, &extra, &p) != Success)
+	                       &real, &format, &n, &extra, &p) != Success)
 		return -1;
 	if (n != 0 && format == 32)
-		result = *(long *)p;
+		result = *(long*) p;
 	XFree(p);
 	return result;
 }
 
 int
-gettextprop(Window w, Atom atom, char *text, unsigned int size)
+gettextprop(Window w, Atom atom, char* text, unsigned int size)
 {
-	char **list = NULL;
+	char** list = NULL;
 	int n;
 	XTextProperty name;
 
@@ -918,7 +963,7 @@ gettextprop(Window w, Atom atom, char *t
 	if (!XGetTextProperty(dpy, w, &name, atom) || !name.nitems)
 		return 0;
 	if (name.encoding == XA_STRING) {
-		strncpy(text, (char *)name.value, size - 1);
+		strncpy(text, (char*) name.value, size - 1);
 	} else if (XmbTextPropertyToTextList(dpy, &name, &list, &n) >= Success && n > 0 && *list) {
 		strncpy(text, *list, size - 1);
 		XFreeStringList(list);
@@ -929,23 +974,23 @@ gettextprop(Window w, Atom atom, char *t
 }
 
 void
-grabbuttons(Client *c, int focused)
+grabbuttons(Client* c, int focused)
 {
 	updatenumlockmask();
 	{
 		unsigned int i, j;
-		unsigned int modifiers[] = { 0, LockMask, numlockmask, numlockmask|LockMask };
+		unsigned int modifiers[] = { 0, LockMask, numlockmask, numlockmask | LockMask };
 		XUngrabButton(dpy, AnyButton, AnyModifier, c->win);
 		if (!focused)
 			XGrabButton(dpy, AnyButton, AnyModifier, c->win, False,
-				BUTTONMASK, GrabModeSync, GrabModeSync, None, None);
+			            BUTTONMASK, GrabModeSync, GrabModeSync, None, None);
 		for (i = 0; i < LENGTH(buttons); i++)
 			if (buttons[i].click == ClkClientWin)
 				for (j = 0; j < LENGTH(modifiers); j++)
 					XGrabButton(dpy, buttons[i].button,
-						buttons[i].mask | modifiers[j],
-						c->win, False, BUTTONMASK,
-						GrabModeAsync, GrabModeSync, None, None);
+					            buttons[i].mask | modifiers[j],
+					            c->win, False, BUTTONMASK,
+					            GrabModeAsync, GrabModeSync, None, None);
 	}
 }
 
@@ -955,9 +1000,9 @@ grabkeys(void)
 	updatenumlockmask();
 	{
 		unsigned int i, j, k;
-		unsigned int modifiers[] = { 0, LockMask, numlockmask, numlockmask|LockMask };
+		unsigned int modifiers[] = { 0, LockMask, numlockmask, numlockmask | LockMask };
 		int start, end, skip;
-		KeySym *syms;
+		KeySym* syms;
 
 		XUngrabKey(dpy, AnyKey, AnyModifier, root);
 		XDisplayKeycodes(dpy, &start, &end);
@@ -970,15 +1015,15 @@ grabkeys(void)
 				if (keys[i].keysym == syms[(k - start) * skip])
 					for (j = 0; j < LENGTH(modifiers); j++)
 						XGrabKey(dpy, k,
-							 keys[i].mod | modifiers[j],
-							 root, True,
-							 GrabModeAsync, GrabModeAsync);
+						         keys[i].mod | modifiers[j],
+						         root, True,
+						         GrabModeAsync, GrabModeAsync);
 		XFree(syms);
 	}
 }
 
 void
-incnmaster(const Arg *arg)
+incnmaster(const Arg* arg)
 {
 	selmon->nmaster = MAX(selmon->nmaster + arg->i, 0);
 	arrange(selmon);
@@ -986,34 +1031,31 @@ incnmaster(const Arg *arg)
 
 #ifdef XINERAMA
 static int
-isuniquegeom(XineramaScreenInfo *unique, size_t n, XineramaScreenInfo *info)
+isuniquegeom(XineramaScreenInfo* unique, size_t n, XineramaScreenInfo* info)
 {
 	while (n--)
-		if (unique[n].x_org == info->x_org && unique[n].y_org == info->y_org
-		&& unique[n].width == info->width && unique[n].height == info->height)
+		if (unique[n].x_org == info->x_org && unique[n].y_org == info->y_org && unique[n].width == info->width && unique[n].height == info->height)
 			return 0;
 	return 1;
 }
 #endif /* XINERAMA */
 
 void
-keypress(XEvent *e)
+keypress(XEvent* e)
 {
 	unsigned int i;
 	KeySym keysym;
-	XKeyEvent *ev;
+	XKeyEvent* ev;
 
-	ev = &e->xkey;
-	keysym = XKeycodeToKeysym(dpy, (KeyCode)ev->keycode, 0);
+	ev     = &e->xkey;
+	keysym = XKeycodeToKeysym(dpy, (KeyCode) ev->keycode, 0);
 	for (i = 0; i < LENGTH(keys); i++)
-		if (keysym == keys[i].keysym
-		&& CLEANMASK(keys[i].mod) == CLEANMASK(ev->state)
-		&& keys[i].func)
+		if (keysym == keys[i].keysym && CLEANMASK(keys[i].mod) == CLEANMASK(ev->state) && keys[i].func)
 			keys[i].func(&(keys[i].arg));
 }
 
 void
-killclient(const Arg *arg)
+killclient(const Arg* arg)
 {
 	if (!selmon->sel)
 		return;
@@ -1029,24 +1071,24 @@ killclient(const Arg *arg)
 }
 
 void
-manage(Window w, XWindowAttributes *wa)
+manage(Window w, XWindowAttributes* wa)
 {
 	Client *c, *t = NULL;
 	Window trans = None;
 	XWindowChanges wc;
 
-	c = ecalloc(1, sizeof(Client));
+	c      = ecalloc(1, sizeof(Client));
 	c->win = w;
 	/* geometry */
 	c->x = c->oldx = wa->x;
 	c->y = c->oldy = wa->y;
 	c->w = c->oldw = wa->width;
 	c->h = c->oldh = wa->height;
-	c->oldbw = wa->border_width;
+	c->oldbw       = wa->border_width;
 
 	updatetitle(c);
 	if (XGetTransientForHint(dpy, w, &trans) && (t = wintoclient(trans))) {
-		c->mon = t->mon;
+		c->mon  = t->mon;
 		c->tags = t->tags;
 	} else {
 		c->mon = selmon;
@@ -1057,8 +1099,8 @@ manage(Window w, XWindowAttributes *wa)
 		c->x = c->mon->wx + c->mon->ww - WIDTH(c);
 	if (c->y + HEIGHT(c) > c->mon->wy + c->mon->wh)
 		c->y = c->mon->wy + c->mon->wh - HEIGHT(c);
-	c->x = MAX(c->x, c->mon->wx);
-	c->y = MAX(c->y, c->mon->wy);
+	c->x  = MAX(c->x, c->mon->wx);
+	c->y  = MAX(c->y, c->mon->wy);
 	c->bw = borderpx;
 
 	wc.border_width = c->bw;
@@ -1068,7 +1110,7 @@ manage(Window w, XWindowAttributes *wa)
 	updatewindowtype(c);
 	updatesizehints(c);
 	updatewmhints(c);
-	XSelectInput(dpy, w, EnterWindowMask|FocusChangeMask|PropertyChangeMask|StructureNotifyMask);
+	XSelectInput(dpy, w, EnterWindowMask | FocusChangeMask | PropertyChangeMask | StructureNotifyMask);
 	grabbuttons(c, 0);
 	if (!c->isfloating)
 		c->isfloating = c->oldstate = trans != None || c->isfixed;
@@ -1077,7 +1119,7 @@ manage(Window w, XWindowAttributes *wa)
 	attach(c);
 	attachstack(c);
 	XChangeProperty(dpy, root, netatom[NetClientList], XA_WINDOW, 32, PropModeAppend,
-		(unsigned char *) &(c->win), 1);
+	                (unsigned char*) &(c->win), 1);
 	XMoveResizeWindow(dpy, c->win, c->x + 2 * sw, c->y, c->w, c->h); /* some windows require this */
 	setclientstate(c, NormalState);
 	if (c->mon == selmon)
@@ -1089,9 +1131,9 @@ manage(Window w, XWindowAttributes *wa)
 }
 
 void
-mappingnotify(XEvent *e)
+mappingnotify(XEvent* e)
 {
-	XMappingEvent *ev = &e->xmapping;
+	XMappingEvent* ev = &e->xmapping;
 
 	XRefreshKeyboardMapping(ev);
 	if (ev->request == MappingKeyboard)
@@ -1099,10 +1141,10 @@ mappingnotify(XEvent *e)
 }
 
 void
-maprequest(XEvent *e)
+maprequest(XEvent* e)
 {
 	static XWindowAttributes wa;
-	XMapRequestEvent *ev = &e->xmaprequest;
+	XMapRequestEvent* ev = &e->xmaprequest;
 
 	if (!XGetWindowAttributes(dpy, ev->window, &wa) || wa.override_redirect)
 		return;
@@ -1111,10 +1153,10 @@ maprequest(XEvent *e)
 }
 
 void
-monocle(Monitor *m)
+monocle(Monitor* m)
 {
 	unsigned int n = 0;
-	Client *c;
+	Client* c;
 
 	for (c = m->clients; c; c = c->next)
 		if (ISVISIBLE(c))
@@ -1126,11 +1168,11 @@ monocle(Monitor *m)
 }
 
 void
-motionnotify(XEvent *e)
+motionnotify(XEvent* e)
 {
-	static Monitor *mon = NULL;
-	Monitor *m;
-	XMotionEvent *ev = &e->xmotion;
+	static Monitor* mon = NULL;
+	Monitor* m;
+	XMotionEvent* ev = &e->xmotion;
 
 	if (ev->window != root)
 		return;
@@ -1143,11 +1185,11 @@ motionnotify(XEvent *e)
 }
 
 void
-movemouse(const Arg *arg)
+movemouse(const Arg* arg)
 {
 	int x, y, ocx, ocy, nx, ny;
-	Client *c;
-	Monitor *m;
+	Client* c;
+	Monitor* m;
 	XEvent ev;
 	Time lasttime = 0;
 
@@ -1159,13 +1201,13 @@ movemouse(const Arg *arg)
 	ocx = c->x;
 	ocy = c->y;
 	if (XGrabPointer(dpy, root, False, MOUSEMASK, GrabModeAsync, GrabModeAsync,
-		None, cursor[CurMove]->cursor, CurrentTime) != GrabSuccess)
+	                 None, cursor[CurMove]->cursor, CurrentTime) != GrabSuccess)
 		return;
 	if (!getrootptr(&x, &y))
 		return;
 	do {
-		XMaskEvent(dpy, MOUSEMASK|ExposureMask|SubstructureRedirectMask, &ev);
-		switch(ev.type) {
+		XMaskEvent(dpy, MOUSEMASK | ExposureMask | SubstructureRedirectMask, &ev);
+		switch (ev.type) {
 		case ConfigureRequest:
 		case Expose:
 		case MapRequest:
@@ -1186,8 +1228,7 @@ movemouse(const Arg *arg)
 				ny = selmon->wy;
 			else if (abs((selmon->wy + selmon->wh) - (ny + HEIGHT(c))) < snap)
 				ny = selmon->wy + selmon->wh - HEIGHT(c);
-			if (!c->isfloating && selmon->lt[selmon->sellt]->arrange
-			&& (abs(nx - c->x) > snap || abs(ny - c->y) > snap))
+			if (!c->isfloating && selmon->lt[selmon->sellt]->arrange && (abs(nx - c->x) > snap || abs(ny - c->y) > snap))
 				togglefloating(NULL);
 			if (!selmon->lt[selmon->sellt]->arrange || c->isfloating)
 				resize(c, nx, ny, c->w, c->h, 1);
@@ -1202,15 +1243,16 @@ movemouse(const Arg *arg)
 	}
 }
 
-Client *
-nexttiled(Client *c)
+Client*
+nexttiled(Client* c)
 {
-	for (; c && (c->isfloating || !ISVISIBLE(c)); c = c->next);
+	for (; c && (c->isfloating || !ISVISIBLE(c)); c = c->next)
+		;
 	return c;
 }
 
 void
-pop(Client *c)
+pop(Client* c)
 {
 	detach(c);
 	attach(c);
@@ -1219,22 +1261,23 @@ pop(Client *c)
 }
 
 void
-propertynotify(XEvent *e)
+propertynotify(XEvent* e)
 {
-	Client *c;
+	Client* c;
 	Window trans;
-	XPropertyEvent *ev = &e->xproperty;
+	XPropertyEvent* ev = &e->xproperty;
 
 	if ((ev->window == root) && (ev->atom == XA_WM_NAME))
 		updatestatus();
 	else if (ev->state == PropertyDelete)
 		return; /* ignore */
 	else if ((c = wintoclient(ev->window))) {
-		switch(ev->atom) {
-		default: break;
+		switch (ev->atom) {
+		default:
+			break;
 		case XA_WM_TRANSIENT_FOR:
 			if (!c->isfloating && (XGetTransientForHint(dpy, c->win, &trans)) &&
-				(c->isfloating = (wintoclient(trans)) != NULL))
+			    (c->isfloating = (wintoclient(trans)) != NULL))
 				arrange(c->mon);
 			break;
 		case XA_WM_NORMAL_HINTS:
@@ -1256,53 +1299,57 @@ propertynotify(XEvent *e)
 }
 
 void
-quit(const Arg *arg)
+quit(const Arg* arg)
 {
 	running = 0;
 }
 
-Monitor *
+Monitor*
 recttomon(int x, int y, int w, int h)
 {
 	Monitor *m, *r = selmon;
-	int a, area = 0;
+	int a, area    = 0;
 
 	for (m = mons; m; m = m->next)
 		if ((a = INTERSECT(x, y, w, h, m)) > area) {
 			area = a;
-			r = m;
+			r    = m;
 		}
 	return r;
 }
 
 void
-resize(Client *c, int x, int y, int w, int h, int interact)
+resize(Client* c, int x, int y, int w, int h, int interact)
 {
 	if (applysizehints(c, &x, &y, &w, &h, interact))
 		resizeclient(c, x, y, w, h);
 }
 
 void
-resizeclient(Client *c, int x, int y, int w, int h)
+resizeclient(Client* c, int x, int y, int w, int h)
 {
 	XWindowChanges wc;
 
-	c->oldx = c->x; c->x = wc.x = x;
-	c->oldy = c->y; c->y = wc.y = y;
-	c->oldw = c->w; c->w = wc.width = w;
-	c->oldh = c->h; c->h = wc.height = h;
-	wc.border_width = c->bw;
-	XConfigureWindow(dpy, c->win, CWX|CWY|CWWidth|CWHeight|CWBorderWidth, &wc);
+	c->oldx = c->x;
+	c->x = wc.x = x;
+	c->oldy     = c->y;
+	c->y = wc.y = y;
+	c->oldw     = c->w;
+	c->w = wc.width = w;
+	c->oldh         = c->h;
+	c->h = wc.height = h;
+	wc.border_width  = c->bw;
+	XConfigureWindow(dpy, c->win, CWX | CWY | CWWidth | CWHeight | CWBorderWidth, &wc);
 	configure(c);
 	XSync(dpy, False);
 }
 
 void
-resizemouse(const Arg *arg)
+resizemouse(const Arg* arg)
 {
 	int ocx, ocy, nw, nh;
-	Client *c;
-	Monitor *m;
+	Client* c;
+	Monitor* m;
 	XEvent ev;
 	Time lasttime = 0;
 
@@ -1314,12 +1361,12 @@ resizemouse(const Arg *arg)
 	ocx = c->x;
 	ocy = c->y;
 	if (XGrabPointer(dpy, root, False, MOUSEMASK, GrabModeAsync, GrabModeAsync,
-		None, cursor[CurResize]->cursor, CurrentTime) != GrabSuccess)
+	                 None, cursor[CurResize]->cursor, CurrentTime) != GrabSuccess)
 		return;
 	XWarpPointer(dpy, None, c->win, 0, 0, 0, 0, c->w + c->bw - 1, c->h + c->bw - 1);
 	do {
-		XMaskEvent(dpy, MOUSEMASK|ExposureMask|SubstructureRedirectMask, &ev);
-		switch(ev.type) {
+		XMaskEvent(dpy, MOUSEMASK | ExposureMask | SubstructureRedirectMask, &ev);
+		switch (ev.type) {
 		case ConfigureRequest:
 		case Expose:
 		case MapRequest:
@@ -1332,11 +1379,8 @@ resizemouse(const Arg *arg)
 
 			nw = MAX(ev.xmotion.x - ocx - 2 * c->bw + 1, 1);
 			nh = MAX(ev.xmotion.y - ocy - 2 * c->bw + 1, 1);
-			if (c->mon->wx + nw >= selmon->wx && c->mon->wx + nw <= selmon->wx + selmon->ww
-			&& c->mon->wy + nh >= selmon->wy && c->mon->wy + nh <= selmon->wy + selmon->wh)
-			{
-				if (!c->isfloating && selmon->lt[selmon->sellt]->arrange
-				&& (abs(nw - c->w) > snap || abs(nh - c->h) > snap))
+			if (c->mon->wx + nw >= selmon->wx && c->mon->wx + nw <= selmon->wx + selmon->ww && c->mon->wy + nh >= selmon->wy && c->mon->wy + nh <= selmon->wy + selmon->wh) {
+				if (!c->isfloating && selmon->lt[selmon->sellt]->arrange && (abs(nw - c->w) > snap || abs(nh - c->h) > snap))
 					togglefloating(NULL);
 			}
 			if (!selmon->lt[selmon->sellt]->arrange || c->isfloating)
@@ -1346,7 +1390,8 @@ resizemouse(const Arg *arg)
 	} while (ev.type != ButtonRelease);
 	XWarpPointer(dpy, None, c->win, 0, 0, 0, 0, c->w + c->bw - 1, c->h + c->bw - 1);
 	XUngrabPointer(dpy, CurrentTime);
-	while (XCheckMaskEvent(dpy, EnterWindowMask, &ev));
+	while (XCheckMaskEvent(dpy, EnterWindowMask, &ev))
+		;
 	if ((m = recttomon(c->x, c->y, c->w, c->h)) != selmon) {
 		sendmon(c, m);
 		selmon = m;
@@ -1355,9 +1400,9 @@ resizemouse(const Arg *arg)
 }
 
 void
-restack(Monitor *m)
+restack(Monitor* m)
 {
-	Client *c;
+	Client* c;
 	XEvent ev;
 	XWindowChanges wc;
 
@@ -1368,15 +1413,16 @@ restack(Monitor *m)
 		XRaiseWindow(dpy, m->sel->win);
 	if (m->lt[m->sellt]->arrange) {
 		wc.stack_mode = Below;
-		wc.sibling = m->barwin;
+		wc.sibling    = m->barwin;
 		for (c = m->stack; c; c = c->snext)
 			if (!c->isfloating && ISVISIBLE(c)) {
-				XConfigureWindow(dpy, c->win, CWSibling|CWStackMode, &wc);
+				XConfigureWindow(dpy, c->win, CWSibling | CWStackMode, &wc);
 				wc.sibling = c->win;
 			}
 	}
 	XSync(dpy, False);
-	while (XCheckMaskEvent(dpy, EnterWindowMask, &ev));
+	while (XCheckMaskEvent(dpy, EnterWindowMask, &ev))
+		;
 }
 
 void
@@ -1399,8 +1445,7 @@ scan(void)
 
 	if (XQueryTree(dpy, root, &d1, &d2, &wins, &num)) {
 		for (i = 0; i < num; i++) {
-			if (!XGetWindowAttributes(dpy, wins[i], &wa)
-			|| wa.override_redirect || XGetTransientForHint(dpy, wins[i], &d1))
+			if (!XGetWindowAttributes(dpy, wins[i], &wa) || wa.override_redirect || XGetTransientForHint(dpy, wins[i], &d1))
 				continue;
 			if (wa.map_state == IsViewable || getstate(wins[i]) == IconicState)
 				manage(wins[i], &wa);
@@ -1408,8 +1453,7 @@ scan(void)
 		for (i = 0; i < num; i++) { /* now the transients */
 			if (!XGetWindowAttributes(dpy, wins[i], &wa))
 				continue;
-			if (XGetTransientForHint(dpy, wins[i], &d1)
-			&& (wa.map_state == IsViewable || getstate(wins[i]) == IconicState))
+			if (XGetTransientForHint(dpy, wins[i], &d1) && (wa.map_state == IsViewable || getstate(wins[i]) == IconicState))
 				manage(wins[i], &wa);
 		}
 		if (wins)
@@ -1418,14 +1462,14 @@ scan(void)
 }
 
 void
-sendmon(Client *c, Monitor *m)
+sendmon(Client* c, Monitor* m)
 {
 	if (c->mon == m)
 		return;
 	unfocus(c, 1);
 	detach(c);
 	detachstack(c);
-	c->mon = m;
+	c->mon  = m;
 	c->tags = m->tagset[m->seltags]; /* assign tags of target monitor */
 	attach(c);
 	attachstack(c);
@@ -1436,19 +1480,19 @@ sendmon(Client *c, Monitor *m)
 }
 
 void
-setclientstate(Client *c, long state)
+setclientstate(Client* c, long state)
 {
 	long data[] = { state, None };
 
 	XChangeProperty(dpy, c->win, wmatom[WMState], wmatom[WMState], 32,
-		PropModeReplace, (unsigned char *)data, 2);
+	                PropModeReplace, (unsigned char*) data, 2);
 }
 
 int
-sendevent(Client *c, Atom proto)
+sendevent(Client* c, Atom proto)
 {
 	int n;
-	Atom *protocols;
+	Atom* protocols;
 	int exists = 0;
 	XEvent ev;
 
@@ -1458,62 +1502,62 @@ sendevent(Client *c, Atom proto)
 		XFree(protocols);
 	}
 	if (exists) {
-		ev.type = ClientMessage;
-		ev.xclient.window = c->win;
+		ev.type                 = ClientMessage;
+		ev.xclient.window       = c->win;
 		ev.xclient.message_type = wmatom[WMProtocols];
-		ev.xclient.format = 32;
-		ev.xclient.data.l[0] = proto;
-		ev.xclient.data.l[1] = CurrentTime;
+		ev.xclient.format       = 32;
+		ev.xclient.data.l[0]    = proto;
+		ev.xclient.data.l[1]    = CurrentTime;
 		XSendEvent(dpy, c->win, False, NoEventMask, &ev);
 	}
 	return exists;
 }
 
 void
-setfocus(Client *c)
+setfocus(Client* c)
 {
 	if (!c->neverfocus)
 		XSetInputFocus(dpy, c->win, RevertToPointerRoot, CurrentTime);
 	XChangeProperty(dpy, root, netatom[NetActiveWindow], XA_WINDOW, 32,
-		PropModeReplace, (unsigned char *)&c->win, 1);
+	                PropModeReplace, (unsigned char*) &c->win, 1);
 	sendevent(c, wmatom[WMTakeFocus]);
 }
 
 void
-setfullscreen(Client *c, int fullscreen)
+setfullscreen(Client* c, int fullscreen)
 {
 	if (fullscreen && !c->isfullscreen) {
 		XChangeProperty(dpy, c->win, netatom[NetWMState], XA_ATOM, 32,
-			PropModeReplace, (unsigned char*)&netatom[NetWMFullscreen], 1);
+		                PropModeReplace, (unsigned char*) &netatom[NetWMFullscreen], 1);
 		c->isfullscreen = 1;
-		c->oldstate = c->isfloating;
-		c->oldbw = c->bw;
-		c->bw = 0;
-		c->isfloating = 1;
+		c->oldstate     = c->isfloating;
+		c->oldbw        = c->bw;
+		c->bw           = 0;
+		c->isfloating   = 1;
 		resizeclient(c, c->mon->mx, c->mon->my, c->mon->mw, c->mon->mh);
 		XRaiseWindow(dpy, c->win);
-	} else if (!fullscreen && c->isfullscreen){
+	} else if (!fullscreen && c->isfullscreen) {
 		XChangeProperty(dpy, c->win, netatom[NetWMState], XA_ATOM, 32,
-			PropModeReplace, (unsigned char*)0, 0);
+		                PropModeReplace, (unsigned char*) 0, 0);
 		c->isfullscreen = 0;
-		c->isfloating = c->oldstate;
-		c->bw = c->oldbw;
-		c->x = c->oldx;
-		c->y = c->oldy;
-		c->w = c->oldw;
-		c->h = c->oldh;
+		c->isfloating   = c->oldstate;
+		c->bw           = c->oldbw;
+		c->x            = c->oldx;
+		c->y            = c->oldy;
+		c->w            = c->oldw;
+		c->h            = c->oldh;
 		resizeclient(c, c->x, c->y, c->w, c->h);
 		arrange(c->mon);
 	}
 }
 
 void
-setlayout(const Arg *arg)
+setlayout(const Arg* arg)
 {
 	if (!arg || !arg->v || arg->v != selmon->lt[selmon->sellt])
 		selmon->sellt ^= 1;
 	if (arg && arg->v)
-		selmon->lt[selmon->sellt] = (Layout *)arg->v;
+		selmon->lt[selmon->sellt] = (Layout*) arg->v;
 	strncpy(selmon->ltsymbol, selmon->lt[selmon->sellt]->symbol, sizeof selmon->ltsymbol);
 	if (selmon->sel)
 		arrange(selmon);
@@ -1523,7 +1567,7 @@ setlayout(const Arg *arg)
 
 /* arg > 1.0 will set mfact absolutely */
 void
-setmfact(const Arg *arg)
+setmfact(const Arg* arg)
 {
 	float f;
 
@@ -1546,45 +1590,46 @@ setup(void)
 
 	/* do not transform children into zombies when they terminate */
 	sigemptyset(&sa.sa_mask);
-	sa.sa_flags = SA_NOCLDSTOP | SA_NOCLDWAIT | SA_RESTART;
+	sa.sa_flags   = SA_NOCLDSTOP | SA_NOCLDWAIT | SA_RESTART;
 	sa.sa_handler = SIG_IGN;
 	sigaction(SIGCHLD, &sa, NULL);
 
 	/* clean up any zombies (inherited from .xinitrc etc) immediately */
-	while (waitpid(-1, NULL, WNOHANG) > 0);
+	while (waitpid(-1, NULL, WNOHANG) > 0)
+		;
 
 	/* init screen */
 	screen = DefaultScreen(dpy);
-	sw = DisplayWidth(dpy, screen);
-	sh = DisplayHeight(dpy, screen);
-	root = RootWindow(dpy, screen);
-	drw = drw_create(dpy, screen, root, sw, sh);
+	sw     = DisplayWidth(dpy, screen);
+	sh     = DisplayHeight(dpy, screen);
+	root   = RootWindow(dpy, screen);
+	drw    = drw_create(dpy, screen, root, sw, sh);
 	if (!drw_fontset_create(drw, fonts, LENGTH(fonts)))
 		die("no fonts could be loaded.");
 	lrpad = drw->fonts->h;
-	bh = drw->fonts->h + 2;
+	bh    = drw->fonts->h + 2;
 	updategeom();
 	/* init atoms */
-	utf8string = XInternAtom(dpy, "UTF8_STRING", False);
-	wmatom[WMProtocols] = XInternAtom(dpy, "WM_PROTOCOLS", False);
-	wmatom[WMDelete] = XInternAtom(dpy, "WM_DELETE_WINDOW", False);
-	wmatom[WMState] = XInternAtom(dpy, "WM_STATE", False);
-	wmatom[WMTakeFocus] = XInternAtom(dpy, "WM_TAKE_FOCUS", False);
-	netatom[NetActiveWindow] = XInternAtom(dpy, "_NET_ACTIVE_WINDOW", False);
-	netatom[NetSupported] = XInternAtom(dpy, "_NET_SUPPORTED", False);
-	netatom[NetWMName] = XInternAtom(dpy, "_NET_WM_NAME", False);
-	netatom[NetWMState] = XInternAtom(dpy, "_NET_WM_STATE", False);
-	netatom[NetWMCheck] = XInternAtom(dpy, "_NET_SUPPORTING_WM_CHECK", False);
-	netatom[NetWMFullscreen] = XInternAtom(dpy, "_NET_WM_STATE_FULLSCREEN", False);
-	netatom[NetWMWindowType] = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE", False);
+	utf8string                     = XInternAtom(dpy, "UTF8_STRING", False);
+	wmatom[WMProtocols]            = XInternAtom(dpy, "WM_PROTOCOLS", False);
+	wmatom[WMDelete]               = XInternAtom(dpy, "WM_DELETE_WINDOW", False);
+	wmatom[WMState]                = XInternAtom(dpy, "WM_STATE", False);
+	wmatom[WMTakeFocus]            = XInternAtom(dpy, "WM_TAKE_FOCUS", False);
+	netatom[NetActiveWindow]       = XInternAtom(dpy, "_NET_ACTIVE_WINDOW", False);
+	netatom[NetSupported]          = XInternAtom(dpy, "_NET_SUPPORTED", False);
+	netatom[NetWMName]             = XInternAtom(dpy, "_NET_WM_NAME", False);
+	netatom[NetWMState]            = XInternAtom(dpy, "_NET_WM_STATE", False);
+	netatom[NetWMCheck]            = XInternAtom(dpy, "_NET_SUPPORTING_WM_CHECK", False);
+	netatom[NetWMFullscreen]       = XInternAtom(dpy, "_NET_WM_STATE_FULLSCREEN", False);
+	netatom[NetWMWindowType]       = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE", False);
 	netatom[NetWMWindowTypeDialog] = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE_DIALOG", False);
-	netatom[NetClientList] = XInternAtom(dpy, "_NET_CLIENT_LIST", False);
+	netatom[NetClientList]         = XInternAtom(dpy, "_NET_CLIENT_LIST", False);
 	/* init cursors */
 	cursor[CurNormal] = drw_cur_create(drw, XC_left_ptr);
 	cursor[CurResize] = drw_cur_create(drw, XC_sizing);
-	cursor[CurMove] = drw_cur_create(drw, XC_fleur);
+	cursor[CurMove]   = drw_cur_create(drw, XC_fleur);
 	/* init appearance */
-	scheme = ecalloc(LENGTH(colors), sizeof(Clr *));
+	scheme = ecalloc(LENGTH(colors), sizeof(Clr*));
 	for (i = 0; i < LENGTH(colors); i++)
 		scheme[i] = drw_scm_create(drw, colors[i], 3);
 	/* init bars */
@@ -1593,30 +1638,28 @@ setup(void)
 	/* supporting window for NetWMCheck */
 	wmcheckwin = XCreateSimpleWindow(dpy, root, 0, 0, 1, 1, 0, 0, 0);
 	XChangeProperty(dpy, wmcheckwin, netatom[NetWMCheck], XA_WINDOW, 32,
-		PropModeReplace, (unsigned char *) &wmcheckwin, 1);
+	                PropModeReplace, (unsigned char*) &wmcheckwin, 1);
 	XChangeProperty(dpy, wmcheckwin, netatom[NetWMName], utf8string, 8,
-		PropModeReplace, (unsigned char *) "dwm", 3);
+	                PropModeReplace, (unsigned char*) "dwm", 3);
 	XChangeProperty(dpy, root, netatom[NetWMCheck], XA_WINDOW, 32,
-		PropModeReplace, (unsigned char *) &wmcheckwin, 1);
+	                PropModeReplace, (unsigned char*) &wmcheckwin, 1);
 	/* EWMH support per view */
 	XChangeProperty(dpy, root, netatom[NetSupported], XA_ATOM, 32,
-		PropModeReplace, (unsigned char *) netatom, NetLast);
+	                PropModeReplace, (unsigned char*) netatom, NetLast);
 	XDeleteProperty(dpy, root, netatom[NetClientList]);
 	/* select events */
-	wa.cursor = cursor[CurNormal]->cursor;
-	wa.event_mask = SubstructureRedirectMask|SubstructureNotifyMask
-		|ButtonPressMask|PointerMotionMask|EnterWindowMask
-		|LeaveWindowMask|StructureNotifyMask|PropertyChangeMask;
-	XChangeWindowAttributes(dpy, root, CWEventMask|CWCursor, &wa);
+	wa.cursor     = cursor[CurNormal]->cursor;
+	wa.event_mask = SubstructureRedirectMask | SubstructureNotifyMask | ButtonPressMask | PointerMotionMask | EnterWindowMask | LeaveWindowMask | StructureNotifyMask | PropertyChangeMask;
+	XChangeWindowAttributes(dpy, root, CWEventMask | CWCursor, &wa);
 	XSelectInput(dpy, root, wa.event_mask);
 	grabkeys();
 	focus(NULL);
 }
 
 void
-seturgent(Client *c, int urg)
+seturgent(Client* c, int urg)
 {
-	XWMHints *wmh;
+	XWMHints* wmh;
 
 	c->isurgent = urg;
 	if (!(wmh = XGetWMHints(dpy, c->win)))
@@ -1627,7 +1670,7 @@ seturgent(Client *c, int urg)
 }
 
 void
-showhide(Client *c)
+showhide(Client* c)
 {
 	if (!c)
 		return;
@@ -1645,7 +1688,7 @@ showhide(Client *c)
 }
 
 void
-spawn(const Arg *arg)
+spawn(const Arg* arg)
 {
 	struct sigaction sa;
 
@@ -1657,17 +1700,17 @@ spawn(const Arg *arg)
 		setsid();
 
 		sigemptyset(&sa.sa_mask);
-		sa.sa_flags = 0;
+		sa.sa_flags   = 0;
 		sa.sa_handler = SIG_DFL;
 		sigaction(SIGCHLD, &sa, NULL);
 
-		execvp(((char **)arg->v)[0], (char **)arg->v);
-		die("dwm: execvp '%s' failed:", ((char **)arg->v)[0]);
+		execvp(((char**) arg->v)[0], (char**) arg->v);
+		die("dwm: execvp '%s' failed:", ((char**) arg->v)[0]);
 	}
 }
 
 void
-tag(const Arg *arg)
+tag(const Arg* arg)
 {
 	if (selmon->sel && arg->ui & TAGMASK) {
 		selmon->sel->tags = arg->ui & TAGMASK;
@@ -1677,7 +1720,7 @@ tag(const Arg *arg)
 }
 
 void
-tagmon(const Arg *arg)
+tagmon(const Arg* arg)
 {
 	if (!selmon->sel || !mons->next)
 		return;
@@ -1685,12 +1728,13 @@ tagmon(const Arg *arg)
 }
 
 void
-tile(Monitor *m)
+tile(Monitor* m)
 {
 	unsigned int i, n, h, mw, my, ty;
-	Client *c;
+	Client* c;
 
-	for (n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++);
+	for (n = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), n++)
+		;
 	if (n == 0)
 		return;
 
@@ -1701,19 +1745,19 @@ tile(Monitor *m)
 	for (i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
 		if (i < m->nmaster) {
 			h = (m->wh - my) / (MIN(n, m->nmaster) - i);
-			resize(c, m->wx, m->wy + my, mw - (2*c->bw), h - (2*c->bw), 0);
+			resize(c, m->wx, m->wy + my, mw - (2 * c->bw), h - (2 * c->bw), 0);
 			if (my + HEIGHT(c) < m->wh)
 				my += HEIGHT(c);
 		} else {
 			h = (m->wh - ty) / (n - i);
-			resize(c, m->wx + mw, m->wy + ty, m->ww - mw - (2*c->bw), h - (2*c->bw), 0);
+			resize(c, m->wx + mw, m->wy + ty, m->ww - mw - (2 * c->bw), h - (2 * c->bw), 0);
 			if (ty + HEIGHT(c) < m->wh)
 				ty += HEIGHT(c);
 		}
 }
 
 void
-togglebar(const Arg *arg)
+togglebar(const Arg* arg)
 {
 	selmon->showbar = !selmon->showbar;
 	updatebarpos(selmon);
@@ -1722,7 +1766,7 @@ togglebar(const Arg *arg)
 }
 
 void
-togglefloating(const Arg *arg)
+togglefloating(const Arg* arg)
 {
 	if (!selmon->sel)
 		return;
@@ -1731,12 +1775,12 @@ togglefloating(const Arg *arg)
 	selmon->sel->isfloating = !selmon->sel->isfloating || selmon->sel->isfixed;
 	if (selmon->sel->isfloating)
 		resize(selmon->sel, selmon->sel->x, selmon->sel->y,
-			selmon->sel->w, selmon->sel->h, 0);
+		       selmon->sel->w, selmon->sel->h, 0);
 	arrange(selmon);
 }
 
 void
-toggletag(const Arg *arg)
+toggletag(const Arg* arg)
 {
 	unsigned int newtags;
 
@@ -1751,7 +1795,7 @@ toggletag(const Arg *arg)
 }
 
 void
-toggleview(const Arg *arg)
+toggleview(const Arg* arg)
 {
 	unsigned int newtagset = selmon->tagset[selmon->seltags] ^ (arg->ui & TAGMASK);
 
@@ -1763,7 +1807,7 @@ toggleview(const Arg *arg)
 }
 
 void
-unfocus(Client *c, int setfocus)
+unfocus(Client* c, int setfocus)
 {
 	if (!c)
 		return;
@@ -1776,16 +1820,16 @@ unfocus(Client *c, int setfocus)
 }
 
 void
-unmanage(Client *c, int destroyed)
+unmanage(Client* c, int destroyed)
 {
-	Monitor *m = c->mon;
+	Monitor* m = c->mon;
 	XWindowChanges wc;
 
 	detach(c);
 	detachstack(c);
 	if (!destroyed) {
 		wc.border_width = c->oldbw;
-		XGrabServer(dpy); /* avoid race conditions */
+		XGrabServer(dpy);                                  /* avoid race conditions */
 		XSetErrorHandler(xerrordummy);
 		XSelectInput(dpy, c->win, NoEventMask);
 		XConfigureWindow(dpy, c->win, CWBorderWidth, &wc); /* restore border */
@@ -1802,10 +1846,10 @@ unmanage(Client *c, int destroyed)
 }
 
 void
-unmapnotify(XEvent *e)
+unmapnotify(XEvent* e)
 {
-	Client *c;
-	XUnmapEvent *ev = &e->xunmap;
+	Client* c;
+	XUnmapEvent* ev = &e->xunmap;
 
 	if ((c = wintoclient(ev->window))) {
 		if (ev->send_event)
@@ -1818,19 +1862,19 @@ unmapnotify(XEvent *e)
 void
 updatebars(void)
 {
-	Monitor *m;
+	Monitor* m;
 	XSetWindowAttributes wa = {
 		.override_redirect = True,
 		.background_pixmap = ParentRelative,
-		.event_mask = ButtonPressMask|ExposureMask
+		.event_mask        = ButtonPressMask | ExposureMask
 	};
-	XClassHint ch = {"dwm", "dwm"};
+	XClassHint ch = { "dwm", "dwm" };
 	for (m = mons; m; m = m->next) {
 		if (m->barwin)
 			continue;
 		m->barwin = XCreateWindow(dpy, root, m->wx, m->by, m->ww, bh, 0, DefaultDepth(dpy, screen),
-				CopyFromParent, DefaultVisual(dpy, screen),
-				CWOverrideRedirect|CWBackPixmap|CWEventMask, &wa);
+		                          CopyFromParent, DefaultVisual(dpy, screen),
+		                          CWOverrideRedirect | CWBackPixmap | CWEventMask, &wa);
 		XDefineCursor(dpy, m->barwin, cursor[CurNormal]->cursor);
 		XMapRaised(dpy, m->barwin);
 		XSetClassHint(dpy, m->barwin, &ch);
@@ -1838,7 +1882,7 @@ updatebars(void)
 }
 
 void
-updatebarpos(Monitor *m)
+updatebarpos(Monitor* m)
 {
 	m->wy = m->my;
 	m->wh = m->mh;
@@ -1853,15 +1897,15 @@ updatebarpos(Monitor *m)
 void
 updateclientlist(void)
 {
-	Client *c;
-	Monitor *m;
+	Client* c;
+	Monitor* m;
 
 	XDeleteProperty(dpy, root, netatom[NetClientList]);
 	for (m = mons; m; m = m->next)
 		for (c = m->clients; c; c = c->next)
 			XChangeProperty(dpy, root, netatom[NetClientList],
-				XA_WINDOW, 32, PropModeAppend,
-				(unsigned char *) &(c->win), 1);
+			                XA_WINDOW, 32, PropModeAppend,
+			                (unsigned char*) &(c->win), 1);
 }
 
 int
@@ -1872,12 +1916,13 @@ updategeom(void)
 #ifdef XINERAMA
 	if (XineramaIsActive(dpy)) {
 		int i, j, n, nn;
-		Client *c;
-		Monitor *m;
-		XineramaScreenInfo *info = XineramaQueryScreens(dpy, &nn);
-		XineramaScreenInfo *unique = NULL;
+		Client* c;
+		Monitor* m;
+		XineramaScreenInfo* info   = XineramaQueryScreens(dpy, &nn);
+		XineramaScreenInfo* unique = NULL;
 
-		for (n = 0, m = mons; m; m = m->next, n++);
+		for (n = 0, m = mons; m; m = m->next, n++)
+			;
 		/* only consider unique geometries as separate screens */
 		unique = ecalloc(nn, sizeof(XineramaScreenInfo));
 		for (i = 0, j = 0; i < nn; i++)
@@ -1888,18 +1933,16 @@ updategeom(void)
 
 		/* new monitors if nn > n */
 		for (i = n; i < nn; i++) {
-			for (m = mons; m && m->next; m = m->next);
+			for (m = mons; m && m->next; m = m->next)
+				;
 			if (m)
 				m->next = createmon();
 			else
 				mons = createmon();
 		}
 		for (i = 0, m = mons; i < nn && m; m = m->next, i++)
-			if (i >= n
-			|| unique[i].x_org != m->mx || unique[i].y_org != m->my
-			|| unique[i].width != m->mw || unique[i].height != m->mh)
-			{
-				dirty = 1;
+			if (i >= n || unique[i].x_org != m->mx || unique[i].y_org != m->my || unique[i].width != m->mw || unique[i].height != m->mh) {
+				dirty  = 1;
 				m->num = i;
 				m->mx = m->wx = unique[i].x_org;
 				m->my = m->wy = unique[i].y_org;
@@ -1909,9 +1952,10 @@ updategeom(void)
 			}
 		/* removed monitors if n > nn */
 		for (i = nn; i < n; i++) {
-			for (m = mons; m && m->next; m = m->next);
+			for (m = mons; m && m->next; m = m->next)
+				;
 			while ((c = m->clients)) {
-				dirty = 1;
+				dirty      = 1;
 				m->clients = c->next;
 				detachstack(c);
 				c->mon = mons;
@@ -1925,11 +1969,11 @@ updategeom(void)
 		free(unique);
 	} else
 #endif /* XINERAMA */
-	{ /* default monitor setup */
+	{  /* default monitor setup */
 		if (!mons)
 			mons = createmon();
 		if (mons->mw != sw || mons->mh != sh) {
-			dirty = 1;
+			dirty    = 1;
 			mons->mw = mons->ww = sw;
 			mons->mh = mons->wh = sh;
 			updatebarpos(mons);
@@ -1946,20 +1990,19 @@ void
 updatenumlockmask(void)
 {
 	unsigned int i, j;
-	XModifierKeymap *modmap;
+	XModifierKeymap* modmap;
 
 	numlockmask = 0;
-	modmap = XGetModifierMapping(dpy);
+	modmap      = XGetModifierMapping(dpy);
 	for (i = 0; i < 8; i++)
 		for (j = 0; j < modmap->max_keypermod; j++)
-			if (modmap->modifiermap[i * modmap->max_keypermod + j]
-				== XKeysymToKeycode(dpy, XK_Num_Lock))
+			if (modmap->modifiermap[i * modmap->max_keypermod + j] == XKeysymToKeycode(dpy, XK_Num_Lock))
 				numlockmask = (1 << i);
 	XFreeModifiermap(modmap);
 }
 
 void
-updatesizehints(Client *c)
+updatesizehints(Client* c)
 {
 	long msize;
 	XSizeHints size;
@@ -1994,11 +2037,11 @@ updatesizehints(Client *c)
 	} else
 		c->minw = c->minh = 0;
 	if (size.flags & PAspect) {
-		c->mina = (float)size.min_aspect.y / size.min_aspect.x;
-		c->maxa = (float)size.max_aspect.x / size.max_aspect.y;
+		c->mina = (float) size.min_aspect.y / size.min_aspect.x;
+		c->maxa = (float) size.max_aspect.x / size.max_aspect.y;
 	} else
 		c->maxa = c->mina = 0.0;
-	c->isfixed = (c->maxw && c->maxh && c->maxw == c->minw && c->maxh == c->minh);
+	c->isfixed    = (c->maxw && c->maxh && c->maxw == c->minw && c->maxh == c->minh);
 	c->hintsvalid = 1;
 }
 
@@ -2006,12 +2049,12 @@ void
 updatestatus(void)
 {
 	if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
-		strcpy(stext, "dwm-"VERSION);
+		strcpy(stext, "dwm-" VERSION);
 	drawbar(selmon);
 }
 
 void
-updatetitle(Client *c)
+updatetitle(Client* c)
 {
 	if (!gettextprop(c->win, netatom[NetWMName], c->name, sizeof c->name))
 		gettextprop(c->win, XA_WM_NAME, c->name, sizeof c->name);
@@ -2020,7 +2063,7 @@ updatetitle(Client *c)
 }
 
 void
-updatewindowtype(Client *c)
+updatewindowtype(Client* c)
 {
 	Atom state = getatomprop(c, netatom[NetWMState]);
 	Atom wtype = getatomprop(c, netatom[NetWMWindowType]);
@@ -2032,9 +2075,9 @@ updatewindowtype(Client *c)
 }
 
 void
-updatewmhints(Client *c)
+updatewmhints(Client* c)
 {
-	XWMHints *wmh;
+	XWMHints* wmh;
 
 	if ((wmh = XGetWMHints(dpy, c->win))) {
 		if (c == selmon->sel && wmh->flags & XUrgencyHint) {
@@ -2051,7 +2094,7 @@ updatewmhints(Client *c)
 }
 
 void
-view(const Arg *arg)
+view(const Arg* arg)
 {
 	if ((arg->ui & TAGMASK) == selmon->tagset[selmon->seltags])
 		return;
@@ -2062,11 +2105,11 @@ view(const Arg *arg)
 	arrange(selmon);
 }
 
-Client *
+Client*
 wintoclient(Window w)
 {
-	Client *c;
-	Monitor *m;
+	Client* c;
+	Monitor* m;
 
 	for (m = mons; m; m = m->next)
 		for (c = m->clients; c; c = c->next)
@@ -2075,12 +2118,12 @@ wintoclient(Window w)
 	return NULL;
 }
 
-Monitor *
+Monitor*
 wintomon(Window w)
 {
 	int x, y;
-	Client *c;
-	Monitor *m;
+	Client* c;
+	Monitor* m;
 
 	if (w == root && getrootptr(&x, &y))
 		return recttomon(x, y, 1, 1);
@@ -2096,25 +2139,17 @@ wintomon(Window w)
  * ignored (especially on UnmapNotify's). Other types of errors call Xlibs
  * default error handler, which may call exit. */
 int
-xerror(Display *dpy, XErrorEvent *ee)
+xerror(Display* dpy, XErrorEvent* ee)
 {
-	if (ee->error_code == BadWindow
-	|| (ee->request_code == X_SetInputFocus && ee->error_code == BadMatch)
-	|| (ee->request_code == X_PolyText8 && ee->error_code == BadDrawable)
-	|| (ee->request_code == X_PolyFillRectangle && ee->error_code == BadDrawable)
-	|| (ee->request_code == X_PolySegment && ee->error_code == BadDrawable)
-	|| (ee->request_code == X_ConfigureWindow && ee->error_code == BadMatch)
-	|| (ee->request_code == X_GrabButton && ee->error_code == BadAccess)
-	|| (ee->request_code == X_GrabKey && ee->error_code == BadAccess)
-	|| (ee->request_code == X_CopyArea && ee->error_code == BadDrawable))
+	if (ee->error_code == BadWindow || (ee->request_code == X_SetInputFocus && ee->error_code == BadMatch) || (ee->request_code == X_PolyText8 && ee->error_code == BadDrawable) || (ee->request_code == X_PolyFillRectangle && ee->error_code == BadDrawable) || (ee->request_code == X_PolySegment && ee->error_code == BadDrawable) || (ee->request_code == X_ConfigureWindow && ee->error_code == BadMatch) || (ee->request_code == X_GrabButton && ee->error_code == BadAccess) || (ee->request_code == X_GrabKey && ee->error_code == BadAccess) || (ee->request_code == X_CopyArea && ee->error_code == BadDrawable))
 		return 0;
 	fprintf(stderr, "dwm: fatal error: request code=%d, error code=%d\n",
-		ee->request_code, ee->error_code);
+	        ee->request_code, ee->error_code);
 	return xerrorxlib(dpy, ee); /* may call exit */
 }
 
 int
-xerrordummy(Display *dpy, XErrorEvent *ee)
+xerrordummy(Display* dpy, XErrorEvent* ee)
 {
 	return 0;
 }
@@ -2122,16 +2157,16 @@ xerrordummy(Display *dpy, XErrorEvent *e
 /* Startup Error handler to check if another window manager
  * is already running. */
 int
-xerrorstart(Display *dpy, XErrorEvent *ee)
+xerrorstart(Display* dpy, XErrorEvent* ee)
 {
 	die("dwm: another window manager is already running");
 	return -1;
 }
 
 void
-zoom(const Arg *arg)
+zoom(const Arg* arg)
 {
-	Client *c = selmon->sel;
+	Client* c = selmon->sel;
 
 	if (!selmon->lt[selmon->sellt]->arrange || !c || c->isfloating)
 		return;
@@ -2141,10 +2176,10 @@ zoom(const Arg *arg)
 }
 
 int
-main(int argc, char *argv[])
+main(int argc, char* argv[])
 {
 	if (argc == 2 && !strcmp("-v", argv[1]))
-		die("dwm-"VERSION);
+		die("dwm-" VERSION);
 	else if (argc != 1)
 		die("usage: dwm [-v]");
 	if (!setlocale(LC_CTYPE, "") || !XSupportsLocale())
--- a/transient.c
+++ b/transient.c
@@ -5,8 +5,10 @@
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 
-int main(void) {
-	Display *d;
+int
+main(void)
+{
+	Display* d;
 	Window r, f, t = None;
 	XSizeHints h;
 	XEvent e;
@@ -16,9 +18,9 @@ int main(void) {
 		exit(1);
 	r = DefaultRootWindow(d);
 
-	f = XCreateSimpleWindow(d, r, 100, 100, 400, 400, 0, 0, 0);
+	f           = XCreateSimpleWindow(d, r, 100, 100, 400, 400, 0, 0, 0);
 	h.min_width = h.max_width = h.min_height = h.max_height = 400;
-	h.flags = PMinSize | PMaxSize;
+	h.flags                                                 = PMinSize | PMaxSize;
 	XSetWMNormalHints(d, f, &h);
 	XStoreName(d, f, "floating");
 	XMapWindow(d, f);
--- a/util.c
+++ b/util.c
@@ -8,7 +8,7 @@
 #include "util.h"
 
 void
-die(const char *fmt, ...)
+die(const char* fmt, ...)
 {
 	va_list ap;
 	int saved_errno;
@@ -19,17 +19,17 @@ die(const char *fmt, ...)
 	vfprintf(stderr, fmt, ap);
 	va_end(ap);
 
-	if (fmt[0] && fmt[strlen(fmt)-1] == ':')
+	if (fmt[0] && fmt[strlen(fmt) - 1] == ':')
 		fprintf(stderr, " %s", strerror(saved_errno));
 	fputc('\n', stderr);
 
 	exit(1);
 }
 
-void *
+void*
 ecalloc(size_t nmemb, size_t size)
 {
-	void *p;
+	void* p;
 
 	if (!(p = calloc(nmemb, size)))
 		die("calloc:");
--- a/util.h
+++ b/util.h
@@ -5,5 +5,5 @@
 #define BETWEEN(X, A, B)        ((A) <= (X) && (X) <= (B))
 #define LENGTH(X)               (sizeof (X) / sizeof (X)[0])
 
-void die(const char *fmt, ...);
-void *ecalloc(size_t nmemb, size_t size);
+void die(const char* fmt, ...);
+void* ecalloc(size_t nmemb, size_t size);
