--- a/dwm.c
+++ b/dwm.c
@@ -1683,6 +1683,7 @@ sendmon(Client* c, Monitor* m)
 {
 	if (c->mon == m)
 		return;
+	int hadfocus = (c == selmon->sel);
 	unfocus(c, 1);
 	detach(c);
 	detachstack(c);
@@ -1694,7 +1695,12 @@ sendmon(Client* c, Monitor* m)
 	if (c->isfullscreen)
 		resizeclient(c, m->mx, m->my, m->mw, m->mh);
 	focus(NULL);
-	arrange(NULL);
+	arrange(m);
+	if (hadfocus) {
+		focus(c);
+		restack(m);
+	} else
+		focus(NULL);
 }
 
 void
